{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "---------- /software/OSutils/modules-4.4.1/share/Modules/modulefiles -----------\n",
      "1SynChro/1.15                    kat/2.3.1\n",
      "3d-dna/180922                    kat/2.4.1\n",
      "A5-miseq/20160825                kentUtils/302.1\n",
      "abyss/2.0.2                      khmer/2.0\n",
      "abyss/2.1.0                      KMA/1.2.4\n",
      "abyss/2.1.1                      KMC/3.0.0\n",
      "abyss/2.1.2                      kneaddata/0.6.1\n",
      "abyss/2.1.3                      kraken/0.10.5\n",
      "abyss/2.1.4                      kraken/1.0\n",
      "abyss/2.1.5                      kraken/1.1\n",
      "adapterremoval/2.2.2             Krona/2.7\n",
      "AfterQC/0.9.1                    LACHESIS/20150929\n",
      "AfterQC/0.9.5                    last/770\n",
      "allpathslg/44837                 lastz/1.02.00\n",
      "allpathslg/44913                 LDhat/81596e2\n",
      "allpathslg/49722                 links/v1.8.5\n",
      "allpathslg/50191                 links/v1.8.6\n",
      "allpathslg/52488                 LocHap/2.0\n",
      "amos/3.1.0                       luarocks/2.4.1\n",
      "angsd/0.913                      lumpy-sv/0.2.13\n",
      "angsd/0.917                      MACS2/2.1.1\n",
      "angsd/0.917-116-g5d087b2         mafft/7.307\n",
      "angsd/0.918                      mandalorion-episode-ii/6219d58\n",
      "angsd/0.919                      mapcomp/1.0\n",
      "angsd/0.921                      mapgd/d3edee2\n",
      "angsd/0.922                      MAPGD/0.4.26\n",
      "angsd/0.923                      MAPGD/0.5\n",
      "angsd/0.925                      mason/0.1.2\n",
      "angsd-wrapper/7224db2            MaSuRCA/3.2.2\n",
      "anytag/2.5.2                     MaSuRCA/3.2.3\n",
      "apsim/7.9-r4047                  MaSuRCA/3.2.4\n",
      "apsim/7.9-r4047-docker           MaSuRCA/3.2.8\n",
      "apsim/7.9-r4176                  MaSuRCA/3.3.3\n",
      "apsimx/2018.01.30.2253           mauve/2015.02.13\n",
      "apsimx/2018.09.28.3099           mcl/14-137\n",
      "apsimx/2019.01.08.3392           mcscan/0.8\n",
      "apsimx/2019.04.03.3693           MECAT/3898797\n",
      "apsimx/2019.06.05.3920           megahit/1.1.1\n",
      "apsimx/2019.07.18.4025           megahit/1.1.3\n",
      "apsimx/2019.10.04.4236           megahit/1.2.6\n",
      "apsimx/2189-docker               meme/4.11.2\n",
      "arcs/1.0.6                       Meraculous/2.2.4\n",
      "art/MountRainier                 Metaassembler/1.5\n",
      "aspera-cli/3.7.2.354.010c3b8     metabat/0.32.4\n",
      "assemblathon_stats/14dfdab       metabat2/2.15\n",
      "assemblathon_stats/2011_10_13    metabat2/2.15-3-g367a7ef\n",
      "asub/2.1                         metaphlan2/2.6.0\n",
      "augustus/3.2.2                   microbiome_helper/4330e63\n",
      "augustus/3.3                     mikado/1.1\n",
      "augustus/3.3.1                   miniasm/0.2\n",
      "augustus/3.3.3-new               miniasm/55cf018\n",
      "bam2fastx/1.3.0                  miniconda2/4.3.13\n",
      "bambam/1.4                       minimap2/2.17\n",
      "bamtools/2.3.0                   minimap2/2.9\n",
      "bamtools/2.4.0                   mira/4.0.2\n",
      "bats/0.4.0                       miRanda/3.3a\n",
      "BBMap/37.28                      mitoz/2.3\n",
      "BBMap/37.53                      mplus/0.1\n",
      "BBMap/37.93                      mrbayes/3.2.7\n",
      "BBMap/38.33                      mrbayes/3.2.7a\n",
      "bcftools/1.3.1                   mrbayes/3.2.7a-gpu\n",
      "bcftools/1.6                     msms/3.2rc_163\n",
      "bcftools/1.9                     multiqc/1.0\n",
      "bcgTree/1.0.8                    multiqc/1.2\n",
      "beagle/4.0                       multiqc/1.3\n",
      "beagle/4.1                       multiqc/1.5\n",
      "BEAST/1.8.4                      multiqc/1.6\n",
      "BEAST2/2.4.7                     multiqc/1.7\n",
      "bedtools/2.20.1                  MUMmer/3.23\n",
      "bedtools/2.21.0                  muscle/3.8.1551\n",
      "bedtools/2.27.1                  muscle/4.0.170\n",
      "bfc-submitted/v1                 nanopack/1.0.0\n",
      "biobambam/0.0.191                ncbi-blast/2.2.25\n",
      "biobambam2/2.0.106               ncbi-blast/2.2.30\n",
      "biobambam2/2.0.44                ncbi-blast/2.6.0\n",
      "bismark/0.19.1                   NetLogo/6.0.4\n",
      "bismark/0.20.0                   newbler/2.5.3\n",
      "blasr/5.2.badc092                newbler/2.7\n",
      "blast/2.2.26                     newbler/2.9\n",
      "BLAT/36                          nextclip/v1.3.1\n",
      "blobtools/1.0                    nextgenmap/v0.5.5\n",
      "boost/1_61_0                     NextGenMap/v0.5.5\n",
      "boost/1_61_0-dev                 ngmlr/0.2.7\n",
      "boost/1_65_1                     NovoGraph/1.0.0\n",
      "boost/1_65_1-dev                 NOVOPlasty/2.7.2\n",
      "bowtie/1.0.0                     NPStat/v1\n",
      "bowtie2/2.2.5                    nQuire/a990a88\n",
      "bowtie2/2.2.9                    NSEG/2018-05-30\n",
      "bowtie2/2.3.4.3                  NxTrim/0.4.2\n",
      "BRAKER/1.11                      NxTrim/v0.4.1-0f17575\n",
      "BRAKER/1.9                       oases/0.2.09\n",
      "BRAKER/2.0.1                     obitools/1.2.11\n",
      "BRAKER/2.0.4                     octave/4.2.2\n",
      "BRAKER/2.0.5                     octave/4.4.1\n",
      "BRAKER/2.1.0                     opendronemap/0.4.0\n",
      "BRAKER/2.1.2                     opendronemap/0.4.1\n",
      "breakdancer/4e44b43              openlava/3.2\n",
      "BSseeker2/2.1.5                  openmpi/1.8.8\n",
      "BUSCO/v1.2                       openmpi/4.0.2\n",
      "BUSCO/v2.0                       OPERA-LG/v2.0.6\n",
      "BUSCO/v3.0.2                     orthofinder/2.3.1\n",
      "bwa/0.7.12                       orthomcl/2.0.9\n",
      "bwa/0.7.15                       PAML/4.9i\n",
      "bwa/0.7.17                       pandaseq/2.10\n",
      "canu/1.3                         pandoc/1.19.2\n",
      "canu/1.5                         PASA/2.0.2\n",
      "canu/1.6                         PASA/2.1.0\n",
      "canu/1.7                         PASA/2.2.0\n",
      "cd-hit/4.6.1                     PASA/2.3.3\n",
      "cdo/1.7.0                        patman/1.2.2\n",
      "cdo/1.9.3                        pblat/2.0\n",
      "cdo/1.9.5                        pblat/2.1\n",
      "cegma/v2.4.010312                PBSIM/308622d\n",
      "cegma/v2.5                       PBSuite/15.8.24\n",
      "checkm/1.0.10                    PBSuite/17.05.09-dbrowneup\n",
      "checkm/1.0.11                    pcl/1.8.1\n",
      "checkm/1.0.12                    pcl/1.9.0\n",
      "checkm/1.0.13                    pcl/1.9.1\n",
      "checkm/1.0.6                     PedigreeSim/v2.0\n",
      "checkm/1.0.7                     perl/5.10.1\n",
      "checkm/1.0.8                     perl/5.10.1-thread-multi\n",
      "chromonomer/1.07                 perl/5.12.5\n",
      "chromosomer/0.1.3                perl/5.12.5-thread-multi\n",
      "cndsrc/2010.10.11                perl/5.14.4\n",
      "cndsrc/2017.11.30                perl/5.14.4-thread-multi\n",
      "cnvnator/0.3.3                   perl/5.16.3\n",
      "cnvnator/0.4.1                   perl/5.16.3-thread-multi\n",
      "CodingQuarry/2.0                 perl/5.18.4\n",
      "COGNATE/v1.0                     perl/5.20.3\n",
      "colordiff/1.0.18                 perl/5.20.3-thread-multi\n",
      "conda/4.5.1                      perl/5.22.2\n",
      "cookiecutter/1.0.0               perl/5.22.2-thread-multi\n",
      "crema/173e9a4                    perl/5.24.0\n",
      "crisp/c6cd86e                    perl/5.24.0-thread-multi\n",
      "crossmap/0.3.3                   perl/5.24.1\n",
      "csvkit/1.0.2                     perl/5.24.1-thread-multi\n",
      "cufflinks/2.2.1                  perl/5.26.0\n",
      "cutadapt/1.10                    perl/5.26.0-thread-multi\n",
      "cutadapt/1.15                    perl/5.28.0\n",
      "dammit/1.2                       perl/5.28.0-thread-multi\n",
      "data-registry/1.0                perl/5.8.9\n",
      "data-registry/1.0-dev            perl/5.8.9-thread-multi\n",
      "data-registry/1.0-test           perl/bio-extras\n",
      "deepvariant/0.7.2-py27h5d9141f_1 perl/bio-extras-5.24.0-legacy\n",
      "diamond/0.9.24                   perl/core\n",
      "Disco/1.1-beta                   perl/ensembl\n",
      "docker-compose/1.11.2            perl/jupyterhub\n",
      "docker-compose/1.7.1             perlbrew/0.76\n",
      "DWGSIM/0561546                   pfr-python2/2.7.13\n",
      "ea-utils/1.04.807                pfr-python3/3.6.1\n",
      "ea-utils/1.1.2-484               pfr-python3/3.6.5\n",
      "ea-utils/1.1.2-537               pfr-python3/3.6.6\n",
      "ea-utils/1.1.2-806               phylip/3.696\n",
      "eddypro-engine/5.1.1             phylobayes/4.1c\n",
      "eddypro-engine/5.2.0             PhyloSift/1.0.1\n",
      "eddypro-engine/5.2.1             picard-tools/1.112\n",
      "eddypro-engine/6.0.0             picard-tools/1.79\n",
      "eddypro-engine/6.1.0             picard-tools/2.10.1\n",
      "eddypro-engine/6.2.0             picard-tools/2.18.7\n",
      "eddypro-engine/6.2.1             picard-tools/2.2.4\n",
      "EffectorP/1.0                    picard-tools/2.9.4\n",
      "eg-all/91                        picrust2/2.3.0\n",
      "eg-web-bacteria/91               pilon/1.18\n",
      "eg-web-common/91                 pilon/1.20\n",
      "eg-web-fungi/91                  pilon/1.23\n",
      "eg-web-metazoa/91                pinfish/0.1.0\n",
      "eg-web-plants/91                 platanus/1.2.4\n",
      "eg-web-protists/91               platanus_trim/1.0.7\n",
      "eg-web-search/91                 Platypus/0.8.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eg-web-testsuite/91              plink/1.07\n",
      "ehive/2.4                        plink/1.90b6.5\n",
      "eigensoft/7.2.1                  plink2/v2.00a2lm\n",
      "eman2/2.3                        plncpro/1.1\n",
      "emboss/6.4.0                     popoolation2/1201\n",
      "emboss/6.5.7                     porechop/0.2.3\n",
      "emboss/6.6.0                     portcullis/1.1.0\n",
      "emirge/0.61.1                    portcullis/1.1.1\n",
      "ensembl/69                       portcullis/1.1.2\n",
      "ensembl/91                       powerPlant/core\n",
      "ensembl/92                       powerPlant/locate\n",
      "ensembl/93                       pplacer/v1.1.alpha17-6-g5cecf99\n",
      "ensembl-analysis/69              ppsPCP/1.0\n",
      "ensembl-analysis/91              primer3/2.2.3\n",
      "ensembl-analysis/92              primer3/2.3.5\n",
      "ensembl-analysis/93              prinseq-lite/0.20.4\n",
      "ensembl-api/69                   Prodigal/2.6.3\n",
      "ensembl-api/91                   proj/5.2.0\n",
      "ensembl-api/92                   prokka/1.11\n",
      "ensembl-api/93                   prokka/1.13\n",
      "ensembl-compara/69               purge_haplotigs/1.1.0\n",
      "ensembl-compara/91               qiime/1.9.1\n",
      "ensembl-compara/92               qiime2/2018.11\n",
      "ensembl-compara/93               qiime2/2018.2\n",
      "ensembl-funcgen/69               qiime2/2019.1\n",
      "ensembl-funcgen/91               qiime2/2019.10\n",
      "ensembl-funcgen/92               qiime2/2019.1-picrust2\n",
      "ensembl-funcgen/93               qiime2/2019.4\n",
      "ensembl-genebuild/69             qiime2/2019.7\n",
      "ensembl-genebuild/91             qiime2/2019.7-picrust2\n",
      "ensembl-genebuild/92             qualimap/v2.2.1\n",
      "ensembl-genebuild/93             quast/4.2\n",
      "ensemblgenomes-api/91            R/3.3.0\n",
      "ensembl-git-tools/d55068a        R/3.3.1\n",
      "ensembl-hive/69                  R/3.3.3\n",
      "ensembl-hive/91                  R/3.4.0\n",
      "ensembl-hive/92                  R/3.4.1\n",
      "ensembl-hive/93                  R/3.4.2\n",
      "ensembl-hive-openlava/69         R/3.4.3\n",
      "ensembl-hive-openlava/91         R/3.4.4\n",
      "ensembl-hive-openlava/92         R/3.5.0\n",
      "ensembl-hive-openlava/93         R/3.5.2\n",
      "ensembl-io/69                    R/3.5.3\n",
      "ensembl-io/91                    R/3.6.1\n",
      "ensembl-io/92                    racon/1.3.0\n",
      "ensembl-io/93                    racon/1.3.1\n",
      "ensembl-killlist/69              racon/1.3.2\n",
      "ensembl-killlist/91              racon/1.3.3\n",
      "ensembl-killlist/92              racon/1.4.0\n",
      "ensembl-killlist/93              racon/1.4.2\n",
      "ensembl-metadata/69              racon/1.4.3\n",
      "ensembl-metadata/91              ragoo/1.01\n",
      "ensembl-metadata/92              ragoo/1.02\n",
      "ensembl-metadata/93              RAILS/v1.2\n",
      "ensembl-notebook/latest          RANBOW/1\n",
      "ensembl-orm/69                   RANBOW/1-temp\n",
      "ensembl-orm/91                   RECON/1.08\n",
      "ensembl-orm/92                   REPCLASS/1.0.1\n",
      "ensembl-orm/93                   RepeatMasker-open/3-3-0\n",
      "ensembl-pipeline/69              RepeatMasker-open/4-0-3\n",
      "ensembl-pipeline/91              RepeatMasker-open/4-0-5\n",
      "ensembl-pipeline/92              RepeatModeler/1.0.8\n",
      "ensembl-pipeline/93              RepeatModeler-open/1.0.11\n",
      "ensembl-production/69            RepeatScout/1.0.5\n",
      "ensembl-production/91            REPET/2.5\n",
      "ensembl-production/92            rnacocktail/0.2.1\n",
      "ensembl-production/93            RSEM/1.3.0\n",
      "ensembl-taxonomy/69              rtg-tools/3.7.1-eb13bbb\n",
      "ensembl-taxonomy/91              ruby/2.1.9\n",
      "ensembl-taxonomy/92              ruby/2.2.7\n",
      "ensembl-taxonomy/93              ruby/2.3.4\n",
      "ensembl-tools/69                 ruby/2.4.1\n",
      "ensembl-tools/91                 sabre/1.000\n",
      "ensembl-tools/92                 Salmon/0.8.2\n",
      "ensembl-tools/93                 Salmon/0.9.1\n",
      "ensembl-variation/69             salsa/v2.2\n",
      "ensembl-variation/91             samtools/0.1.18\n",
      "ensembl-variation/92             samtools/0.1.19\n",
      "ensembl-variation/93             samtools/1.1\n",
      "ensembl-vep/69                   samtools/1.2\n",
      "ensembl-vep/91                   samtools/1.3.1\n",
      "ensembl-vep/92                   samtools/1.7\n",
      "ensembl-vep/93                   samtools/1.9\n",
      "e-PCR/2.3.9                      ScanIndel/1.2dev\n",
      "ericscript/0.5.5                 SDhap/56d7678\n",
      "e-utilities/5.40                 seqkit/0.7.0\n",
      "EVidenceModeler/1.1.1            seqtk/1.2\n",
      "evigene/02-11-2017               sga/0.10.10-2\n",
      "evigene/18-01-2018               sga/0.10.13\n",
      "evigene/18-09-2013               sga/0.10.15\n",
      "exonerate/2.4.0                  sga/0.9.19\n",
      "FALCON/1.7.4                     sga/0.9.35\n",
      "fasta/35.4.12                    SGSGeneLoss/0.1\n",
      "fasta/36.3.8g                    singularity/2.4.2\n",
      "fast-gbs/2016.12.14              singularity/2.5.1\n",
      "fast-gbs/2017.01.12              singularity/3\n",
      "fastp/0.12.5                     SLiM/3.1\n",
      "fastp/0.19.4                     smalt/0.7.6\n",
      "fastp/0.20.0                     snakemake/5.10.0\n",
      "FastQC/0.11.1                    snap/1.0beta.18\n",
      "FastQC/0.11.2                    snapr/0.16alpha.51\n",
      "FastQC/0.11.7                    sniffles/1.0.11\n",
      "fastq_screen/v0.11.1             SnpEff/4.3t\n",
      "fastq_screen/v0.14.0             SNPhylo/20140701\n",
      "fastq_screen/v0.4.4              SNPhylo/20160204\n",
      "fastq_screen/v0.5.2              SOAPdenovo/1.05\n",
      "faststructure/1.0                SOAPdenovo/2.04\n",
      "FastTree/2.1.10                  SOAPdenovo-Trans/1.03\n",
      "fastx_toolkit/0.0.13             SolexaQA/3.1.5\n",
      "FGMP/1.0                         sortmerna/2.1\n",
      "FLASH/1.2.11                     sortmerna/3.0.3\n",
      "flye/2.4.2                       SPAdes/3.10.1\n",
      "flye/2.5                         SPAdes/3.11.1\n",
      "freebayes/1.1.0                  SPAdes/3.12.0\n",
      "freebayes/1.3.1                  SPAdes/3.13.0\n",
      "freebayes/v1.0.2-16-gd466dde     SPAdes/3.14.0\n",
      "freetype/2.9                     SPAdes/3.7.1\n",
      "GapFiller/v1.10                  SqueezeMeta/0.4.4\n",
      "GARM/0.7.4                       SqueezeMeta/1.0.0-beta\n",
      "gatk/1.0                         sratoolkit/2.8.2.1\n",
      "gatk/3.7                         SSPACE/2.0\n",
      "gatk/3.8.0                       stacks/1.40\n",
      "Gblocks/0.91b                    stacks/1.40_withbam\n",
      "gdal/2.4.0                       stacks/2.0\n",
      "GEM/20130406-045632              stacks/2.1\n",
      "GeMoMa/1.4.2                     stacks/2.2\n",
      "geneid/v1.4.4                    stamp/2.1.3\n",
      "GeneMark/4.21                    standard-RAxML/8.2.11\n",
      "GeneMark/4.33                    standard-RAxML/8.2.9\n",
      "GeneMark/4.56                    STAR/2.4.1d\n",
      "genomethreader/1.7.0             STAR/2.4.2a\n",
      "genometools/1.5.10               STAR/2.5.2a\n",
      "getorganelle/1.6.2e              STAR/2.5.2b\n",
      "gffread/0.9.6                    STAR/2.5.3a\n",
      "git/2.11.0                       STAR/2.6.1a\n",
      "git/2.12.2                       STAR/2.6.1d\n",
      "git/2.14.1                       STAR-Fusion/0.8.0\n",
      "git/2.21.0                       stringtie/1.3.3b.Linux_x86_64\n",
      "git-annex/6.20160613-g35dbe35    subread/1.4.5\n",
      "gitflow/15aab26                  subread/1.5.3\n",
      "git-lfs/2.3.0                    supernova/1.2.2\n",
      "globus-cli/1.2.2                 survivor/1.0.5\n",
      "globus-cli/1.6.0                 SV2/1.4.3.2\n",
      "gmap/2016-06-09                  svaba/1.1.3\n",
      "gmap/2017-06-20                  svim/1.2.0\n",
      "gmap/2017-09-30                  svtools/0.3.2\n",
      "gmap/2018-03-01                  swan/3516c2f\n",
      "gmap/2018-03-25                  tabix/0.2.6\n",
      "gmap/2018-05-11                  TargQC/1.4.4\n",
      "gmap/2018-05-30                  tasr/1.6.2\n",
      "gmato/v1.2                       tassel/3.0\n",
      "GMcloser/1.6.2                   tassel/4.0\n",
      "go/1.9                           tassel/5.0\n",
      "GroIMP/1.5                       tassel/5.2.28\n",
      "gromacs/2018                     taxmaps/0.2.1\n",
      "gromacs/2018.1                   texlive/20151117\n",
      "gromacs/2018.2                   tigmint/1.1.2\n",
      "gromacs/2018.3                   tophat/2.0.13\n",
      "gromacs/2018.4                   tophat/2.1.1\n",
      "gromacs/2018.5                   totalview/2017.2.11\n",
      "gromacs/2019                     TransDecoder/3.0.1\n",
      "guppy/3.2.4                      TransDecoder/5.0.1\n",
      "guppy/3.4.4                      transrate/1.0.3\n",
      "hagfish/05d08d1                  trf/4.07\n",
      "hagfish/1_May_2013               trim_galore/0.4.1\n",
      "hapcol/97d4a5e                   trim_galore/0.4.3\n",
      "hapcompass/0.8.2                 Trimmomatic/0.36\n",
      "HapCUT2/0121162                  trinityrnaseq/2.0.6\n",
      "Haplosim/ae652a2                 trinityrnaseq/2.2.0\n",
      "Haplosim/bd39c6f                 trinityrnaseq/2.3.2\n",
      "HapTree/1.0                      trinityrnaseq/2.4.0\n",
      "hdf5/1.8.17                      trinityrnaseq/2.5.1\n",
      "HiC-pro/2.11.1                   trinityrnaseq/2.6.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hicup/0.5.9                      trinotate/3.0.2\n",
      "hicup/0.7.2                      tRNAscan-SE/1.3.1\n",
      "hisat2/2.1.0                     tRNAscan-SE/2.0\n",
      "hmmer/3.0                        ucsc-pslcdnafilter/324\n",
      "hmmer/3.1b1                      unixODBC/2.3.0\n",
      "hmmer/3.1b2                      vcflib/50a3229\n",
      "HTSeq/0.6.1                      vcflib/586c5ae\n",
      "HTSeq/0.7.2                      vcflib/73b7f13\n",
      "HTSeq/0.9.1                      vcftools/0.1.12\n",
      "htslib/1.3.1                     vcftools/0.1.14\n",
      "htslib/1.7                       velvet/1.2.10\n",
      "htslib/1.9                       VersioningService/69\n",
      "hub/2.8.4                        VersioningService/91\n",
      "humann2/0.11.1                   VersioningService/92\n",
      "iCommands/4.1.8                  VersioningService/93\n",
      "idba/a1bafe6                     vg/1.10.0\n",
      "IGV/2.3.93                       vg/1.11.0\n",
      "infernal/1.1.2                   vg/1.12.0\n",
      "iqtree/1.6.11                    vg/1.12.1\n",
      "JAGS/4.0.0                       vg/1.13.0\n",
      "JAGS/4.2.0                       vg/1.7.0\n",
      "java/1.7.0                       vg/1.8.0\n",
      "java/1.8.0                       vg/1.9.0\n",
      "jcvi/0.7.3                       ViennaRNA/2.2.2\n",
      "jcvi/0.7.7                       vsearch/2.3.2\n",
      "jcvi/0.8.12                      wgs/8.3rc2\n",
      "jellyfish/1.1.10                 whatshap/0.14.1-py36_0\n",
      "jellyfish/2.1.3                  wise/2.4.1\n",
      "jellyfish/2.2.10                 wkhtmlox/0.12.3\n",
      "jmodeltest/2.1.10                wtdbg2/2.2\n",
      "jq/1.6\n"
     ]
    }
   ],
   "source": [
    "mkdir ZW_call\n",
    "module avail"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. animal_groups_to_ref.vcf(VCF file with variants per measured base location per trevally individual)\n",
    "\n",
    "Function:\n",
    "bgzip the VCF file, which is a prerequisite to indexing with bcftools.\n",
    "    \n",
    "Output:\n",
    "1. ZW.vcf.gz (VCF file with variants per measured base location per trevally individual in bgzipped form)\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <368068> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "module load bcftools/1.9 \n",
    "vcf=/workspace/hramzr/vcf_file_generation/01.freebayes/animal_groups_to_ref.vcf\n",
    "module load tabix/0.2.6\n",
    "bsub -e ZW_call/error \"bgzip -c $vcf > ZW_call/ZW.vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. ZW.vcf.gz (VCF file with variants per measured base location per trevally individual in bgzipped form)\n",
    "\n",
    "Function:\n",
    "Index the ZW.vcf.gz file, which is a prerequisite to calling VCF subsets with bcftools.\n",
    "    \n",
    "Output:\n",
    "1. ZW.vcf.gz.csi(index file of the ZW.vcf.gz file)\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <368148> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "module load bcftools/1.9 \n",
    "bsub -e index.err \"bcftools index ZW_call/ZW.vcf.gz\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. ZW.vcf.gz (VCF file with variants per measured base location per trevally individual in bgzipped form)\n",
    "\n",
    "\n",
    "Function:\n",
    "Substract subsets at specific genomic locations, this consists of the SDR in scaffolds 001800 and 000374 and 6 \n",
    "total genomic locations of similar size for comparison sake.\n",
    "    \n",
    "Output:\n",
    "1. 8 subset VCF files\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load bcftools/1.9 \n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally001800:1-6000>ZW_call/001800SDR\n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally000475:1-3000>ZW_call/R1_6000size\n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally000376:1-3000>ZW_call/R2_6000size \n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally001701:1-3000>ZW_call/R3_6000size\n",
    "# ----------------\n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally000374:1-3000>ZW_call/000374SDR\n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally000474:1-3000>ZW_call/R1_3000size\n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally000377:1-3000>ZW_call/R2_3000size \n",
    "bcftools view ZW_call/ZW.vcf.gz --regions trevally001700:1-3000>ZW_call/R3_3000size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. c(list of VCF files with their locations of specific genomic regions)\n",
    "\n",
    "Function:\n",
    "Convert the VCF format into a genotype format for downstream analysis and use.\n",
    "    \n",
    "Output:\n",
    "1. CSV files with SNP data in genotype format per genomic region from the input list \"c\"\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "   *****       ***   vcfR   ***       *****\n",
      "   This is vcfR 1.8.0 \n",
      "     browseVignettes('vcfR') # Documentation\n",
      "     citation('vcfR') # Citation\n",
      "   *****       *****      *****       *****\n",
      "\n",
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Attaching package: 'data.table'\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    between, first, last\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 78\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 78\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 78\n",
      "  row_num: 0\n",
      "Processed variant: 78\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 107\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 107\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 107\n",
      "  row_num: 0\n",
      "Processed variant: 107\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 41\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 41\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 41\n",
      "  row_num: 0\n",
      "Processed variant: 41\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 46\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 46\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 46\n",
      "  row_num: 0\n",
      "Processed variant: 46\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 66\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 66\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 66\n",
      "  row_num: 0\n",
      "Processed variant: 66\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 52\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 52\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 52\n",
      "  row_num: 0\n",
      "Processed variant: 52\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 171\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 171\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 171\n",
      "  row_num: 0\n",
      "Processed variant: 171\n",
      "All variants processed\n",
      "Scanning file to determine attributes.\n",
      "File attributes:\n",
      "  meta lines: 2069\n",
      "  header_line: 2070\n",
      "  variant count: 145\n",
      "  column count: 22\n",
      "Meta line 2069 read in.\n",
      "All meta lines processed.\n",
      "gt matrix initialized.\n",
      "Character matrix gt created.\n",
      "  Character matrix gt rows: 145\n",
      "  Character matrix gt cols: 22\n",
      "  skip: 0\n",
      "  nrows: 145\n",
      "  row_num: 0\n",
      "Processed variant: 145\n",
      "All variants processed\n"
     ]
    }
   ],
   "source": [
    "library(vcfR)\n",
    "library(dplyr)\n",
    "library(data.table)\n",
    "for (region in c(\"ZW_call/001800SDR\", \"ZW_call/R1_6000size\", \"ZW_call/R2_6000size\", \"ZW_call/R3_6000size\",\n",
    "     \"ZW_call/000374SDR\", \"ZW_call/R1_3000size\", \"ZW_call/R2_3000size\", \"ZW_call/R3_3000size\")){\n",
    "    vcf <- read.vcfR(region, checkFile=F)\n",
    "    gt <- extract.gt(vcf, element = c('GT'))\n",
    "    ?extract.gt\n",
    "    write.csv(gt, paste(region,\".csv\" ,sep=\"\"))\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. calldir(String variable with directionary path)\n",
    "2. run_list(List with String variables that represent VCF region subsets)\n",
    "\n",
    "Function:\n",
    "Count deletions and hetero/homo zygosity in male and female samples and write this off to a CSV file.\n",
    "    \n",
    "Output:\n",
    "1. CSV file with a count of deletions and hetero/homo zygosity in male and female samples\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Scaffold 001800 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 185\n",
      "Male deletions are: 0\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 8\n",
      "Homozygous male count: 326\n",
      "Homozygous female count: 252\n",
      "Heterozygous male count: 546\n",
      "Heterozygous female count: 283\n",
      "Heterozygous ratio(male/female) 1.9293286219081272\n",
      "Homozygous ratio(male/female) 1.2936507936507937\n",
      "Homo/hetero zygous ratio difference 0.6356778282573334\n",
      "\n",
      "Scaffold R1_600 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 5\n",
      "Male deletions are: 24\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 0\n",
      "Homozygous male count: 478\n",
      "Homozygous female count: 462\n",
      "Heterozygous male count: 725\n",
      "Heterozygous female count: 637\n",
      "Heterozygous ratio(male/female) 1.1381475667189953\n",
      "Homozygous ratio(male/female) 1.0346320346320346\n",
      "Homo/hetero zygous ratio difference 0.10351553208696074\n",
      "\n",
      "Scaffold R2_600 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 3\n",
      "Male deletions are: 1\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 0\n",
      "Homozygous male count: 233\n",
      "Homozygous female count: 186\n",
      "Heterozygous male count: 286\n",
      "Heterozygous female count: 243\n",
      "Heterozygous ratio(male/female) 1.176954732510288\n",
      "Homozygous ratio(male/female) 1.2526881720430108\n",
      "Homo/hetero zygous ratio difference 0.07573343953272271\n",
      "\n",
      "Scaffold R3_600 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 0\n",
      "Male deletions are: 0\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 0\n",
      "Homozygous male count: 270\n",
      "Homozygous female count: 202\n",
      "Heterozygous male count: 322\n",
      "Heterozygous female count: 276\n",
      "Heterozygous ratio(male/female) 1.1666666666666667\n",
      "Homozygous ratio(male/female) 1.3366336633663367\n",
      "Homo/hetero zygous ratio difference 0.16996699669966997\n",
      "\n",
      "Scaffold 000374 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 233\n",
      "Male deletions are: 0\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 3\n",
      "Homozygous male count: 331\n",
      "Homozygous female count: 154\n",
      "Heterozygous male count: 462\n",
      "Heterozygous female count: 163\n",
      "Heterozygous ratio(male/female) 2.834355828220859\n",
      "Homozygous ratio(male/female) 2.1493506493506493\n",
      "Homo/hetero zygous ratio difference 0.6850051788702096\n",
      "\n",
      "Scaffold R1_300 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 6\n",
      "Male deletions are: 4\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 0\n",
      "Homozygous male count: 299\n",
      "Homozygous female count: 232\n",
      "Heterozygous male count: 360\n",
      "Heterozygous female count: 306\n",
      "Heterozygous ratio(male/female) 1.1764705882352942\n",
      "Homozygous ratio(male/female) 1.2887931034482758\n",
      "Homo/hetero zygous ratio difference 0.11232251521298164\n",
      "\n",
      "Scaffold R2_300 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 15\n",
      "Male deletions are: 17\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 0\n",
      "Homozygous male count: 863\n",
      "Homozygous female count: 732\n",
      "Heterozygous male count: 1180\n",
      "Heterozygous female count: 1011\n",
      "Heterozygous ratio(male/female) 1.1671612265084075\n",
      "Homozygous ratio(male/female) 1.1789617486338797\n",
      "Homo/hetero zygous ratio difference 0.011800522125472224\n",
      "\n",
      "Scaffold R3_300 stats \n",
      "-------------------------------------\n",
      "\n",
      "Female deletions are: 2\n",
      "Male deletions are: 0\n",
      "Instances of 7/7 males homozygous and 6/6 females heterozygous 0\n",
      "Instances of 7/7 males heterozygous and 6/6 females homozygous 0\n",
      "Homozygous male count: 820\n",
      "Homozygous female count: 724\n",
      "Heterozygous male count: 1015\n",
      "Heterozygous female count: 868\n",
      "Heterozygous ratio(male/female) 1.1693548387096775\n",
      "Homozygous ratio(male/female) 1.132596685082873\n",
      "Homo/hetero zygous ratio difference 0.03675815362680446\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import csv\n",
    "\n",
    "class Del_SNP_counter:\n",
    "    def __init__(self, csv_name):\n",
    "        self.csv_name = csv_name\n",
    "        self.homo=[\"0/0\", \"1/1\", \"2/2\", \"3/3\", \"4/4\", \"5/5\", \"6/6\", \"7/7\"]\n",
    "        self.female_deletions = 0\n",
    "        self.male_deletions = 0\n",
    "        self.homo_countM = 0\n",
    "        self.hetero_countM = 0\n",
    "        self.homo_countF = 0\n",
    "        self.hetero_countF = 0\n",
    "        self.absoluteMheFho = 0\n",
    "        self.absoluteFheMho = 0\n",
    "        self.ordered_csv = self.reorder()\n",
    "        self.counts()\n",
    "        \n",
    "        \n",
    "    def reorder(self):\n",
    "        #read csv in python\n",
    "        df = pd.read_csv(self.csv_name+\".csv\", index_col=0)\n",
    "        #insert order of male to female\n",
    "        df = df[[\"01_group\", \"03_group\", \"05_group\", \"09_group\",\n",
    "                 \"11_group\", \"12_group\", \"13_group\", \"02_group\",\n",
    "                 \"04_group\", \"06_group\", \"07_group\", \"08_group\", \"10_group\"]]\n",
    "        df = df.applymap(str)\n",
    "        df.to_csv(self.csv_name+\"_ordered.csv\", sep=',', encoding='utf-8')\n",
    "        return self.csv_name+\"_ordered.csv\"\n",
    "\n",
    "    def counts(self):\n",
    "        f = open(self.ordered_csv, \"r\")\n",
    "        f = f.readlines()\n",
    "        for line in f[1:]:\n",
    "            line = line.strip()\n",
    "            line = line.split(\",\")\n",
    "            males = line[1:8]\n",
    "            females = line [8:]\n",
    "            for value in males:\n",
    "                if value in self.homo:\n",
    "                    self.homo_countM += 1\n",
    "                if value == \"nan\":\n",
    "                    self.male_deletions += 1\n",
    "                else:\n",
    "                    self.hetero_countM += 1\n",
    "            for value in females:\n",
    "                if value in self.homo:\n",
    "                    self.homo_countF += 1\n",
    "                if value == \"nan\":\n",
    "                    self.female_deletions += 1\n",
    "                else:\n",
    "                    self.hetero_countF += 1\n",
    "            if females.count(\"1/1\") == 6 and males.count(\"0/1\") == 7:\n",
    "                self.absoluteMheFho += 1\n",
    "            if females.count(\"0/1\") == 6 and males.count(\"1/1\") == 7:\n",
    "                self.absoluteFheMho += 1              \n",
    "\n",
    "    def stats(self, f):\n",
    "        \n",
    "        ratio_diff = str(abs((self.hetero_countM/self.hetero_countF)-(self.homo_countM/self.homo_countF)))\n",
    "        f.write(\"Scaffold \"+self.csv_name[8:14]+\"\\t\"+str(self.female_deletions)+\"\\t\"+str(self.male_deletions)+\"\\t\"\n",
    "                +str(self.absoluteFheMho)+\"\\t\"+str(self.absoluteMheFho)+\"\\t\"+str(self.homo_countM)+\"\\t\"+str(self.homo_countF)\n",
    "                +\"\\t\"+str(self.hetero_countM)+\"\\t\"+str(self.hetero_countF)+\"\\t\"+str(self.hetero_countM/self.hetero_countF)+\"\\t\"\n",
    "                +str(self.homo_countM/self.homo_countF)+\"\\t\"+ratio_diff+\"\\t\")\n",
    "        f.write(\"\\n\")\n",
    "        print (\"\\nScaffold\", self.csv_name[8:14],\"stats \\n-------------------------------------\\n\")\n",
    "        print (\"Female deletions are:\", self.female_deletions)\n",
    "        print (\"Male deletions are:\", self.male_deletions)\n",
    "        print(\"Instances of 7/7 males homozygous and 6/6 females heterozygous\", self.absoluteFheMho)\n",
    "        print(\"Instances of 7/7 males heterozygous and 6/6 females homozygous\", self.absoluteMheFho)\n",
    "        print (\"Homozygous male count:\", self.homo_countM)\n",
    "        print (\"Homozygous female count:\", self.homo_countF)\n",
    "        print (\"Heterozygous male count:\", self.hetero_countM)\n",
    "        print (\"Heterozygous female count:\", self.hetero_countF)\n",
    "        print(\"Heterozygous ratio(male/female)\", self.hetero_countM/self.hetero_countF)\n",
    "        print(\"Homozygous ratio(male/female)\", self.homo_countM/self.homo_countF)\n",
    "        print(\"Homo/hetero zygous ratio difference\", ratio_diff)\n",
    "        \n",
    "def main():\n",
    "    with open('ZW_call/del_snp_stats.csv', 'w') as f:\n",
    "        f.write(\"Scaffold\\tFemale DELS\\tMale DELS\\tAll males hom all females het\"\n",
    "                \"\\tAll males het all females hom\\tHom males\\tHom females\"\n",
    "                \"\\tHet males\\tHet females\\tHet ratio(male/female)\\tHom ratio(male/female)\"\n",
    "                \"\\tRatio difference\\n\")\n",
    "        calldir=\"ZW_call/\"\n",
    "        run_list=[\"001800SDR\", \"R1_6000size\", \"R2_6000size\", \"R3_6000size\", \n",
    "                  \"000374SDR\", \"R1_3000size\", \"R2_3000size\", \"R3_3000size\"]\n",
    "        for iterations in run_list:\n",
    "            run = Del_SNP_counter(calldir+iterations)   \n",
    "            run.stats(f)\n",
    "main()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "PFR Python3-3.6.6",
   "language": "python",
   "name": "pfr-python3-3.6.6"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
