{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "module avail"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loading important variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "files=/input/genomic/fish/Chrysophrys/auratus/Resequencing/CAGRF15217/AGRF_CAGRF15217_H2JYTDMXX/\n",
    "wd=/powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/\n",
    "ids=/input/genomic/fish/Chrysophrys/auratus/Resequencing/CAGRF15217/sample_and_individual_identification.txt\n",
    "reference=/powerplant/output/genomic/fish/Chrysophrys/auratus/Genome/1000.Genome.v.1.0/Chrysophrys_auratus.v.1.0.chromosomes.male.map.fsa\n",
    "reference_index=${reference}.fai\n",
    "sup=/workspace/hramzr/2_Phd_PROJECT/WGS/support_files/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analysing startdata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#look for abberations in data\n",
    "module load FastQC/0.11.7\n",
    "module load multiqc/1.7\n",
    "fqdir=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/FQ/\n",
    "bsub \\\n",
    "-e ${fqdir}FASTQC.e -o ${fqdir}FASTQC.o -J \"FQC\" \\\n",
    "\"fastqc --nogroup -q -t 2 ${files}*.gz -o $fqdir\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "IN=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/FQ/\n",
    "OUT=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/multiqc/\n",
    "module load MultiQC/1.3\n",
    "bsub \\\n",
    "-e ${OUT}FASTQC.e -o ${OUT}FASTQC.o -J \"multiqc\" \\\n",
    "\"multiqc ${IN} -o ${OUT}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Trimming data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Based on FASTQC:\n",
    "* Now we will remove adapters, over-represented sequences, and poor-quality bases from the reads:\n",
    "    * We remove the first 9 bases (biased reads)\n",
    "    * We remove trailing bases if quality is less than 10\n",
    "    * After clipping, the min length for a read will be 75 bp\n",
    "* The `Clip_Seqs.fa` file contains the TruSeq adapter sequences and homo-polymer sequences to clip\n",
    "    * This file needs to be edited to contain the appropriate sequences.\n",
    "* The input for this step are the raw reads\n",
    "* The output are the trimmed reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make list of FQS\n",
    "up=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/unpaired/\n",
    "p=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/\n",
    "files=/input/genomic/fish/Chrysophrys/auratus/Resequencing/CAGRF15217/AGRF_CAGRF15217_H2JYTDMXX/\n",
    "sup=/workspace/hramzr/2_Phd_PROJECT/WGS/support_files/\n",
    "rm ${sup}trimfile\n",
    "ls $files | egrep \".gz\" | awk -F \"_\" '{print $3}' | sort | uniq -c | awk '{print $2}'>${sup}identifiers\n",
    "while read -r id\n",
    "do\n",
    "var1=$(ls $files | egrep \".gz\" | egrep \"_${id}_\" | \\\n",
    "awk '{print \"/input/genomic/fish/Chrysophrys/auratus/Resequencing/CAGRF15217/AGRF_CAGRF15217_H2JYTDMXX/\"$0}' | tr \"\\n\" \" \" )\n",
    "\n",
    "var2=$(echo \"${up}${id}_unpaired_L1R1.fq.gz ${up}${id}_unpaired_L1R2.fq.gz \\\n",
    "${up}${id}_unpaired_L2R1.fq.gz ${up}${id}_unpaired_L2R2.fq.gz ${p}${id}_L1R1.fq.gz ${p}${id}_L1R2.fq.gz \\\n",
    "${p}${id}_L2R1.fq.gz ${p}${id}_L2R2.fq.gz\")\n",
    "echo $var1 $var2>>${sup}trimfile\n",
    "done < ${sup}identifiers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat ${sup}trimfile | awk '{print $3}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "files=/input/genomic/fish/Chrysophrys/auratus/Resequencing/CAGRF15217/AGRF_CAGRF15217_H2JYTDMXX/\n",
    "TRPATH=/software/bioinformatics/Trimmomatic-0.36\n",
    "TRPTH=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/\n",
    "file_dir=${sup}fqlist\n",
    "log=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/log/\n",
    "CLIP=/workspace/hramzr/2_Phd_PROJECT/WGS/adapters/Clip_Seqs.fa\n",
    "sup=/workspace/hramzr/2_Phd_PROJECT/WGS/support_files/\n",
    "\n",
    "while read -r in1 in2 in3 in4 up1 up2 up3 up4 p1 p2 p3 p4\n",
    "do\n",
    "bsub \\\n",
    "-o ${log}trim.o -e ${log}trim.e -J \"triml1\" -n8  \\\n",
    "\"java -jar ${TRPATH}/trimmomatic-0.36.jar \\\n",
    "PE -phred33 \\\n",
    "-threads 8 \\\n",
    "${in1} ${in2} \\\n",
    "${p1} ${up1} ${p2} ${up2} \\\n",
    "ILLUMINACLIP:${CLIP}:2:30:10 \\\n",
    "HEADCROP:9 \\\n",
    "TRAILING:10 \\\n",
    "SLIDINGWINDOW:5:20 \\\n",
    "MINLEN:75\"\n",
    "\n",
    "bsub \\\n",
    "-o ${log}trim.o -e ${log}trim.e -J \"triml2\" -n8  \\\n",
    "\"java -jar ${TRPATH}/trimmomatic-0.36.jar \\\n",
    "PE -phred33 \\\n",
    "-threads 8 \\\n",
    "${in3} ${in4} \\\n",
    "${p3} ${up3} ${p4} ${up4} \\\n",
    "ILLUMINACLIP:${CLIP}:2:30:10 \\\n",
    "HEADCROP:9 \\\n",
    "TRAILING:10 \\\n",
    "SLIDINGWINDOW:5:20 \\\n",
    "MINLEN:75\"\n",
    "done < ${sup}trimfile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat /powerplant/workspace/hramzr/2_Phd_PROJECT/Genomics/Fish/Resequencing/Chrysophrys_auratus/10565/metadata/additional_metadata/sample_and_individual_identification.txt | \\\n",
    "grep -w \"SNF_gDNA_9\"\n",
    "\n",
    "\n",
    "# cat /powerplant/workspace/hramzr/2_Phd_PROJECT/Genomics/Fish/Resequencing/Chrysophrys_auratus/10565/metadata/additional_metadata/sample_and_individual_identification.txt \n",
    "echo ${id}\n",
    "cat /powerplant/workspace/hramzr/2_Phd_PROJECT/Genomics/Fish/Resequencing/Chrysophrys_auratus/10565/metadata/additional_metadata/sample_and_individual_identification.txt | \\\n",
    "grep -w \"SNF_gDNA_${id}}\"\n",
    "\n",
    "cat t | sort | uniq -c | wc -l\n",
    "\n",
    "cat ${sup}identifiers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check trim quality with FASTQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fqdir=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/FQ/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine replicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergedir=/workspace/hramzr/2_Phd_PROJECT/WGS/merging/\n",
    "trimdir=/workspace/hramzr/2_Phd_PROJECT/WGS/trimming/\n",
    "log=/workspace/hramzr/2_Phd_PROJECT/WGS/merging/log/\n",
    "\n",
    "while read -r id\n",
    "do\n",
    "\n",
    "bsub \\\n",
    "-e ${log}r1.e -o ${log}r1.o \\\n",
    "\"cat ${trimdir}${id}_L1R1.fq.gz ${id}_L2R1.fq.gz > ${mergedir}${id}_R1.fq.gz\"\n",
    "\n",
    "bsub \\\n",
    "-e ${log}r2.e -o ${log}r2.o \\\n",
    "\"cat ${trimdir}${id}_L1R2.fq.gz ${id}_L2R2.fq.gz > ${mergedir}${id}_R2.fq.gz\"\n",
    "\n",
    "done < ${sup}identifiers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# mark adapters add metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmpdir=/workspace/hramzr/2_Phd_PROJECT/WGS/MIA/tmp/\n",
    "METRICS=/workspace/hramzr/2_Phd_PROJECT/WGS/MIA/metrics/\n",
    "PICARD=/software/bioinformatics/picard-tools-2.18.7/picard.jar\n",
    "OUT=/workspace/hramzr/2_Phd_PROJECT/WGS/MIA\n",
    "log=/workspace/hramzr/2_Phd_PROJECT/WGS/MIA/log/\n",
    "IN=/workspace/hramzr/2_Phd_PROJECT/WGS/merging/\n",
    "sup=/workspace/hramzr/2_Phd_PROJECT/WGS/support_files/\n",
    "module load bwa/0.7.17\n",
    "module load java\n",
    "rm t\n",
    "while read -r id\n",
    "do\n",
    "new_id=$(cat /powerplant/workspace/hramzr/2_Phd_PROJECT/Genomics/Fish/Resequencing/Chrysophrys_auratus/10565/metadata/additional_metadata/sample_and_individual_identification.txt | \\\n",
    "grep -w \"SNF_gDNA_${id}\" | awk '{print $3}')\n",
    "file=$(ls /input/genomic/fish/Chrysophrys/auratus/Resequencing/CAGRF15217/AGRF_CAGRF15217_H2JYTDMXX/ | egrep \"_${id}_\" | head -n1)\n",
    "NAME=`basename ${file}`\n",
    "PREFIX=`echo ${NAME} | awk -F[_] '{print $1\"_\"$2\"_\"$3\"_\"$4\"_\"$5\"_MIA\"}'`\n",
    "RGID=\"RGID_${new_id}\"\n",
    "SAMPLEID=\"SID_${new_id}\"\n",
    "LIB=`echo ${PREFIX} | awk -F[_] '{print $1\"_\"$2}'`\n",
    "file1=\"${id}_R1.fq.gz\"\n",
    "file2=\"${id}_R2.fq.gz\"\n",
    "echo \"${id}\"\n",
    "bsub \\\n",
    "-o ${log}mia.o -e ${log}mia.e -n 4 -J \"${id}\" \\\n",
    "-R \"rusage[mem=10000] span[hosts=1]\" \\\n",
    "\"java -jar -Xmx48G -Djava.io.tmpdir=${tmpdir} $PICARD FastqToSam \\\n",
    "MAX_RECORDS_IN_RAM=1000000 \\\n",
    "VALIDATION_STRINGENCY=STRICT \\\n",
    "FASTQ=${IN}${file1} \\\n",
    "FASTQ2=${IN}${file2} \\\n",
    "OUTPUT=/dev/stdout \\\n",
    "READ_GROUP_NAME=${RGID} \\\n",
    "SAMPLE_NAME=${SAMPLEID} \\\n",
    "LIBRARY_NAME=${LIB} \\\n",
    "PLATFORM_UNIT=unit1 \\\n",
    "PLATFORM=illumina \\\n",
    "SEQUENCING_CENTER=AGRF \\\n",
    "RUN_DATE=2017-08-15T00:00:00-0400 | \\\n",
    "java -jar -Xmx32G -Djava.io.tmpdir=${tmpdir} $PICARD MarkIlluminaAdapters \\\n",
    "MAX_RECORDS_IN_RAM=1000000 \\\n",
    "VALIDATION_STRINGENCY=STRICT \\\n",
    "I=/dev/stdin \\\n",
    "O=${OUT}/${new_id}.bam \\\n",
    "M=${METRICS}/${new_id}.txt \\\n",
    "ADAPTERS=INDEXED \\\n",
    "ADAPTERS=DUAL_INDEXED \\\n",
    "ADAPTERS=PAIRED_END \\\n",
    "ADAPTERS=SINGLE_END \\\n",
    "ADAPTERS=ALTERNATIVE_SINGLE_END\"\n",
    "done < ${sup}identifiers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run BWA mem"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load bwa/0.7.17\n",
    "OUT=\"/workspace/hramzr/2_Phd_PROJECT/snapper_genome/bwadir/\"\n",
    "LOG=\"/workspace/hramzr/2_Phd_PROJECT/WGS/aligning/log\"\n",
    "COMMAND=\"bwa index -p ${OUT}Chrysophrys_auratus.v.1.0.chromosomes.male.map ${OUT}Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta \\\n",
    "${OUT}Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta\"\n",
    "# echo ${COMMAND}\n",
    "bsub -o ${LOG}/bwa_index.out -e ${LOG}/bwa_index.err \\\n",
    "     -J BWA_INDEX \\\n",
    "     -n 3 \\\n",
    "     -R \"span[hosts=1]\" \\\n",
    "     ${COMMAND}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <398498> is submitted to default queue <lowpriority>.\n",
      "Job <398499> is submitted to default queue <lowpriority>.\n",
      "Job <398500> is submitted to default queue <lowpriority>.\n",
      "Job <398501> is submitted to default queue <lowpriority>.\n",
      "Job <398502> is submitted to default queue <lowpriority>.\n",
      "Job <398503> is submitted to default queue <lowpriority>.\n",
      "Job <398504> is submitted to default queue <lowpriority>.\n",
      "Job <398505> is submitted to default queue <lowpriority>.\n",
      "Job <398506> is submitted to default queue <lowpriority>.\n",
      "Job <398507> is submitted to default queue <lowpriority>.\n",
      "Job <398508> is submitted to default queue <lowpriority>.\n",
      "Job <398509> is submitted to default queue <lowpriority>.\n",
      "Job <398510> is submitted to default queue <lowpriority>.\n",
      "Job <398511> is submitted to default queue <lowpriority>.\n",
      "Job <398512> is submitted to default queue <lowpriority>.\n",
      "Job <398513> is submitted to default queue <lowpriority>.\n",
      "Job <398514> is submitted to default queue <lowpriority>.\n",
      "Job <398515> is submitted to default queue <lowpriority>.\n",
      "Job <398516> is submitted to default queue <lowpriority>.\n",
      "Job <398517> is submitted to default queue <lowpriority>.\n",
      "Job <398518> is submitted to default queue <lowpriority>.\n",
      "Job <398519> is submitted to default queue <lowpriority>.\n",
      "Job <398520> is submitted to default queue <lowpriority>.\n",
      "Job <398521> is submitted to default queue <lowpriority>.\n",
      "Job <398522> is submitted to default queue <lowpriority>.\n",
      "Job <398523> is submitted to default queue <lowpriority>.\n",
      "Job <398524> is submitted to default queue <lowpriority>.\n",
      "Job <398525> is submitted to default queue <lowpriority>.\n",
      "Job <398526> is submitted to default queue <lowpriority>.\n",
      "Job <398527> is submitted to default queue <lowpriority>.\n",
      "Job <398528> is submitted to default queue <lowpriority>.\n",
      "Job <398529> is submitted to default queue <lowpriority>.\n",
      "Job <398530> is submitted to default queue <lowpriority>.\n",
      "Job <398531> is submitted to default queue <lowpriority>.\n",
      "Job <398532> is submitted to default queue <lowpriority>.\n",
      "Job <398533> is submitted to default queue <lowpriority>.\n",
      "Job <398534> is submitted to default queue <lowpriority>.\n",
      "Job <398535> is submitted to default queue <lowpriority>.\n",
      "Job <398536> is submitted to default queue <lowpriority>.\n",
      "Job <398537> is submitted to default queue <lowpriority>.\n",
      "Job <398538> is submitted to default queue <lowpriority>.\n",
      "Job <398539> is submitted to default queue <lowpriority>.\n",
      "Job <398540> is submitted to default queue <lowpriority>.\n",
      "Job <398541> is submitted to default queue <lowpriority>.\n",
      "Job <398542> is submitted to default queue <lowpriority>.\n",
      "Job <398543> is submitted to default queue <lowpriority>.\n",
      "Job <398544> is submitted to default queue <lowpriority>.\n",
      "Job <398545> is submitted to default queue <lowpriority>.\n",
      "Job <398546> is submitted to default queue <lowpriority>.\n",
      "Job <398547> is submitted to default queue <lowpriority>.\n",
      "Job <398548> is submitted to default queue <lowpriority>.\n",
      "Job <398549> is submitted to default queue <lowpriority>.\n",
      "Job <398550> is submitted to default queue <lowpriority>.\n",
      "Job <398551> is submitted to default queue <lowpriority>.\n",
      "Job <398552> is submitted to default queue <lowpriority>.\n",
      "Job <398553> is submitted to default queue <lowpriority>.\n",
      "Job <398554> is submitted to default queue <lowpriority>.\n",
      "Job <398555> is submitted to default queue <lowpriority>.\n",
      "Job <398556> is submitted to default queue <lowpriority>.\n",
      "Job <398557> is submitted to default queue <lowpriority>.\n",
      "Job <398558> is submitted to default queue <lowpriority>.\n",
      "Job <398559> is submitted to default queue <lowpriority>.\n",
      "Job <398560> is submitted to default queue <lowpriority>.\n",
      "Job <398561> is submitted to default queue <lowpriority>.\n",
      "Job <398562> is submitted to default queue <lowpriority>.\n",
      "Job <398563> is submitted to default queue <lowpriority>.\n",
      "Job <398564> is submitted to default queue <lowpriority>.\n",
      "Job <398565> is submitted to default queue <lowpriority>.\n",
      "Job <398566> is submitted to default queue <lowpriority>.\n",
      "Job <398567> is submitted to default queue <lowpriority>.\n",
      "Job <398568> is submitted to default queue <lowpriority>.\n",
      "Job <398569> is submitted to default queue <lowpriority>.\n",
      "Job <398570> is submitted to default queue <lowpriority>.\n",
      "Job <398571> is submitted to default queue <lowpriority>.\n",
      "Job <398572> is submitted to default queue <lowpriority>.\n",
      "Job <398573> is submitted to default queue <lowpriority>.\n",
      "Job <398574> is submitted to default queue <lowpriority>.\n",
      "Job <398575> is submitted to default queue <lowpriority>.\n",
      "Job <398576> is submitted to default queue <lowpriority>.\n",
      "Job <398577> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "module load bwa/0.7.17\n",
    "IN=\"/workspace/hramzr/2_Phd_PROJECT/WGS/MIA\"\n",
    "OUT=\"/workspace/hramzr/2_Phd_PROJECT/WGS/aligning\"\n",
    "LOG=\"${OUT}/log\"\n",
    "OUT_BWA=\"/workspace/hramzr/2_Phd_PROJECT/WGS/out_bwa\"\n",
    "BAMS=`ls ${IN}/*.bam`\n",
    "TEMP=\"/workspace/hramzr/2_Phd_PROJECT/WGS/tmp\"\n",
    "ref=\"/workspace/hramzr/2_Phd_PROJECT/snapper_genome/Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta\"\n",
    "PICARD=\"/software/bioinformatics/picard-tools-2.18.7/picard.jar\"\n",
    "fai=\"/workspace/hramzr/2_Phd_PROJECT/snapper_genome/Chrysophrys_auratus.v.1.0.chromosomes.male.map\"\n",
    "JAVA=\"java -jar -Xmx64G\"\n",
    "for bam in ${BAMS}\n",
    "    do\n",
    "      NAME=`basename -s .bam ${bam}`\n",
    "      PREFIX=`echo ${NAME} | awk -F[.] '{print $1}'`\n",
    "#       echo ${NAME}\n",
    "#       echo ${PREFIX}\n",
    "      ### Create Named Pipes: ###\n",
    "      R1=\"${TEMP}/${PREFIX}_1\"\n",
    "      R2=\"${TEMP}/${PREFIX}_2\"\n",
    "      rm -f ${R1}\n",
    "      rm -f ${R2}\n",
    "      mkfifo ${R1}\n",
    "      mkfifo ${R2}\n",
    "#       echo ${R1}\n",
    "#       echo ${R2}\n",
    "      BWA=\"bwa mem -t 32 -M ${ref} ${R1} ${R2}\"\n",
    "      COMMAND=\"${JAVA} ${PICARD} SamToFastq \\\n",
    "                  I=${bam} \\\n",
    "                  F=${R1} \\\n",
    "                  F2=${R2} \\\n",
    "                  CLIPPING_ATTRIBUTE=XT \\\n",
    "                  CLIPPING_ACTION=2 \\\n",
    "                  INTERLEAVE=false \\\n",
    "                  NON_PF=true \\\n",
    "                  TMP_DIR=${TEMP} | \\\n",
    "               ${BWA} | \\\n",
    "               tee ${OUT_BWA}/${PREFIX}_aligned.bam | \\\n",
    "               ${JAVA} ${PICARD} MergeBamAlignment \\\n",
    "                  ALIGNED_BAM=/dev/stdin \\\n",
    "                  UNMAPPED_BAM=${bam} \\\n",
    "                  OUTPUT=${OUT}/${PREFIX}_MBA.bam \\\n",
    "                  R=${ref} \\\n",
    "                  ADD_MATE_CIGAR=true \\\n",
    "                  CLIP_ADAPTERS=true \\\n",
    "                  CLIP_OVERLAPPING_READS=true \\\n",
    "                  INCLUDE_SECONDARY_ALIGNMENTS=true \\\n",
    "                  MAX_INSERTIONS_OR_DELETIONS=-1 \\\n",
    "                  PRIMARY_ALIGNMENT_STRATEGY=BestMapq \\\n",
    "                  ATTRIBUTES_TO_RETAIN=XS \\\n",
    "                  MAX_RECORDS_IN_RAM=1000000 \\\n",
    "                  VALIDATION_STRINGENCY=STRICT \\\n",
    "                  SORT_ORDER=coordinate \\\n",
    "                  CREATE_INDEX=true \\\n",
    "                  TMP_DIR=${TEMP}\"\n",
    "      # echo ${COMMAND}\n",
    "      bsub -o ${LOG}/${PREFIX}.out \\\n",
    "           -e ${LOG}/${PREFIX}.err \\\n",
    "           -J MBA \\\n",
    "           -n 6 \\\n",
    "           -R \"rusage[mem=100000] span[hosts=1]\" \\\n",
    "           ${COMMAND} \n",
    "    done\n",
    "\n",
    "### MBA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mark duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "000475368\n",
      "Job <398742> is submitted to default queue <lowpriority>.\n",
      "008801547\n",
      "Job <398743> is submitted to default queue <lowpriority>.\n",
      "008808514\n",
      "Job <398744> is submitted to default queue <lowpriority>.\n",
      "008813278\n",
      "Job <398745> is submitted to default queue <lowpriority>.\n",
      "008819521\n",
      "Job <398746> is submitted to default queue <lowpriority>.\n",
      "008820870\n",
      "Job <398747> is submitted to default queue <lowpriority>.\n",
      "008821357\n",
      "Job <398748> is submitted to default queue <lowpriority>.\n",
      "008825059\n",
      "Job <398749> is submitted to default queue <lowpriority>.\n",
      "008825377\n",
      "Job <398750> is submitted to default queue <lowpriority>.\n",
      "008828555\n",
      "Job <398751> is submitted to default queue <lowpriority>.\n",
      "008828871\n",
      "Job <398752> is submitted to default queue <lowpriority>.\n",
      "008831557\n",
      "Job <398753> is submitted to default queue <lowpriority>.\n",
      "008831607\n",
      "Job <398754> is submitted to default queue <lowpriority>.\n",
      "008833125\n",
      "Job <398755> is submitted to default queue <lowpriority>.\n",
      "008841043\n",
      "Job <398756> is submitted to default queue <lowpriority>.\n",
      "008843785\n",
      "Job <398757> is submitted to default queue <lowpriority>.\n",
      "009022893\n",
      "Job <398758> is submitted to default queue <lowpriority>.\n",
      "009026369\n",
      "Job <398759> is submitted to default queue <lowpriority>.\n",
      "009042121\n",
      "Job <398760> is submitted to default queue <lowpriority>.\n",
      "012317803\n",
      "Job <398761> is submitted to default queue <lowpriority>.\n",
      "014513579\n",
      "Job <398762> is submitted to default queue <lowpriority>.\n",
      "015275636\n",
      "Job <398763> is submitted to default queue <lowpriority>.\n",
      "027033489\n",
      "Job <398764> is submitted to default queue <lowpriority>.\n",
      "077793299\n",
      "Job <398765> is submitted to default queue <lowpriority>.\n",
      "083800743\n",
      "Job <398766> is submitted to default queue <lowpriority>.\n",
      "090001781\n",
      "Job <398767> is submitted to default queue <lowpriority>.\n",
      "096100681\n",
      "Job <398768> is submitted to default queue <lowpriority>.\n",
      "109947113\n",
      "Job <398769> is submitted to default queue <lowpriority>.\n",
      "140942470\n",
      "Job <398770> is submitted to default queue <lowpriority>.\n",
      "145693354\n",
      "Job <398771> is submitted to default queue <lowpriority>.\n",
      "184889037\n",
      "Job <398772> is submitted to default queue <lowpriority>.\n",
      "197458872\n",
      "Job <398773> is submitted to default queue <lowpriority>.\n",
      "212582674\n",
      "Job <398774> is submitted to default queue <lowpriority>.\n",
      "220020828\n",
      "Job <398775> is submitted to default queue <lowpriority>.\n",
      "225909887\n",
      "Job <398776> is submitted to default queue <lowpriority>.\n",
      "239454473\n",
      "Job <398777> is submitted to default queue <lowpriority>.\n",
      "243417191\n",
      "Job <398778> is submitted to default queue <lowpriority>.\n",
      "261931510\n",
      "Job <398779> is submitted to default queue <lowpriority>.\n",
      "262447031\n",
      "Job <398780> is submitted to default queue <lowpriority>.\n",
      "275378212\n",
      "Job <398781> is submitted to default queue <lowpriority>.\n",
      "279192515\n",
      "Job <398782> is submitted to default queue <lowpriority>.\n",
      "296841509\n",
      "Job <398783> is submitted to default queue <lowpriority>.\n",
      "299826357\n",
      "Job <398784> is submitted to default queue <lowpriority>.\n",
      "301050525\n",
      "Job <398785> is submitted to default queue <lowpriority>.\n",
      "305981370\n",
      "Job <398786> is submitted to default queue <lowpriority>.\n",
      "322378787\n",
      "Job <398787> is submitted to default queue <lowpriority>.\n",
      "358075192\n",
      "Job <398788> is submitted to default queue <lowpriority>.\n",
      "390130722\n",
      "Job <398789> is submitted to default queue <lowpriority>.\n",
      "396985257\n",
      "Job <398790> is submitted to default queue <lowpriority>.\n",
      "397065822\n",
      "Job <398791> is submitted to default queue <lowpriority>.\n",
      "410128477\n",
      "Job <398792> is submitted to default queue <lowpriority>.\n",
      "453817879\n",
      "Job <398793> is submitted to default queue <lowpriority>.\n",
      "477198320\n",
      "Job <398794> is submitted to default queue <lowpriority>.\n",
      "498476286\n",
      "Job <398795> is submitted to default queue <lowpriority>.\n",
      "504243687\n",
      "Job <398796> is submitted to default queue <lowpriority>.\n",
      "520095148\n",
      "Job <398797> is submitted to default queue <lowpriority>.\n",
      "551118489\n",
      "Job <398798> is submitted to default queue <lowpriority>.\n",
      "592865129\n",
      "Job <398799> is submitted to default queue <lowpriority>.\n",
      "600570777\n",
      "Job <398800> is submitted to default queue <lowpriority>.\n",
      "629742264\n",
      "Job <398801> is submitted to default queue <lowpriority>.\n",
      "636039763\n",
      "Job <398802> is submitted to default queue <lowpriority>.\n",
      "650616751\n",
      "Job <398803> is submitted to default queue <lowpriority>.\n",
      "661305928\n",
      "Job <398804> is submitted to default queue <lowpriority>.\n",
      "690179560\n",
      "Job <398805> is submitted to default queue <lowpriority>.\n",
      "734562795\n",
      "Job <398806> is submitted to default queue <lowpriority>.\n",
      "750067070\n",
      "Job <398807> is submitted to default queue <lowpriority>.\n",
      "753129955\n",
      "Job <398808> is submitted to default queue <lowpriority>.\n",
      "764636580\n",
      "Job <398809> is submitted to default queue <lowpriority>.\n",
      "808141137\n",
      "Job <398810> is submitted to default queue <lowpriority>.\n",
      "808177555\n",
      "Job <398811> is submitted to default queue <lowpriority>.\n",
      "817081985\n",
      "Job <398812> is submitted to default queue <lowpriority>.\n",
      "844410657\n",
      "Job <398813> is submitted to default queue <lowpriority>.\n",
      "853067240\n",
      "Job <398814> is submitted to default queue <lowpriority>.\n",
      "859394261\n",
      "Job <398815> is submitted to default queue <lowpriority>.\n",
      "909956485\n",
      "Job <398816> is submitted to default queue <lowpriority>.\n",
      "934689216\n",
      "Job <398817> is submitted to default queue <lowpriority>.\n",
      "967038331\n",
      "Job <398818> is submitted to default queue <lowpriority>.\n",
      "985661182\n",
      "Job <398819> is submitted to default queue <lowpriority>.\n",
      "986994303\n",
      "Job <398820> is submitted to default queue <lowpriority>.\n",
      "998916898\n",
      "Job <398821> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "in=/workspace/hramzr/2_Phd_PROJECT/WGS/aligning/\n",
    "out=/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates\n",
    "tmpdir=/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/tmp/\n",
    "log=/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/log/\n",
    "PICARD=/software/bioinformatics/picard-tools-2.18.7/picard.jar\n",
    "metrics=/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/metrics/\n",
    "sup=/workspace/hramzr/2_Phd_PROJECT/WGS/support_files/\n",
    "bams=$(ls /workspace/hramzr/2_Phd_PROJECT/WGS/aligning/*.bam)\n",
    "for bam in $bams\n",
    "do\n",
    "NAME=`basename -s _MBA.bam ${bam}`\n",
    "echo $NAME\n",
    "bsub \\\n",
    "-o ${log}mdup.o -e ${log}mdup.e -J \"mdup ${NAME}\" \\\n",
    "-R \"rusage[mem=80000] span[hosts=1]\" \\\n",
    "\"java -jar -Xmx32G -Djava.io.tmpdir=${tmpdir} $PICARD MarkDuplicates \\\n",
    " MAX_FILE_HANDLES=1024 \\\n",
    " I=$bam \\\n",
    " MAX_RECORDS_IN_RAM=500000 \\\n",
    " MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=500000 \\\n",
    " VALIDATION_STRINGENCY=STRICT \\\n",
    " CREATE_INDEX=true \\\n",
    " REMOVE_DUPLICATES=true \\\n",
    " REMOVE_SEQUENCING_DUPLICATES=true \\\n",
    " O=${out}/${NAME}_M.bam \\\n",
    " M=${metrics}${NAME}_metrics.txt\"\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# check stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <399919> is submitted to default queue <lowpriority>.\n",
      "Job <399920> is submitted to default queue <lowpriority>.\n",
      "Job <399921> is submitted to default queue <lowpriority>.\n",
      "Job <399922> is submitted to default queue <lowpriority>.\n",
      "Job <399923> is submitted to default queue <lowpriority>.\n",
      "Job <399924> is submitted to default queue <lowpriority>.\n",
      "Job <399925> is submitted to default queue <lowpriority>.\n",
      "Job <399926> is submitted to default queue <lowpriority>.\n",
      "Job <399927> is submitted to default queue <lowpriority>.\n",
      "Job <399928> is submitted to default queue <lowpriority>.\n",
      "Job <399929> is submitted to default queue <lowpriority>.\n",
      "Job <399930> is submitted to default queue <lowpriority>.\n",
      "Job <399931> is submitted to default queue <lowpriority>.\n",
      "Job <399932> is submitted to default queue <lowpriority>.\n",
      "Job <399933> is submitted to default queue <lowpriority>.\n",
      "Job <399934> is submitted to default queue <lowpriority>.\n",
      "Job <399935> is submitted to default queue <lowpriority>.\n",
      "Job <399936> is submitted to default queue <lowpriority>.\n",
      "Job <399937> is submitted to default queue <lowpriority>.\n",
      "Job <399938> is submitted to default queue <lowpriority>.\n",
      "Job <399939> is submitted to default queue <lowpriority>.\n",
      "Job <399940> is submitted to default queue <lowpriority>.\n",
      "Job <399941> is submitted to default queue <lowpriority>.\n",
      "Job <399942> is submitted to default queue <lowpriority>.\n",
      "Job <399943> is submitted to default queue <lowpriority>.\n",
      "Job <399944> is submitted to default queue <lowpriority>.\n",
      "Job <399945> is submitted to default queue <lowpriority>.\n",
      "Job <399946> is submitted to default queue <lowpriority>.\n",
      "Job <399947> is submitted to default queue <lowpriority>.\n",
      "Job <399948> is submitted to default queue <lowpriority>.\n",
      "Job <399949> is submitted to default queue <lowpriority>.\n",
      "Job <399950> is submitted to default queue <lowpriority>.\n",
      "Job <399951> is submitted to default queue <lowpriority>.\n",
      "Job <399952> is submitted to default queue <lowpriority>.\n",
      "Job <399953> is submitted to default queue <lowpriority>.\n",
      "Job <399954> is submitted to default queue <lowpriority>.\n",
      "Job <399955> is submitted to default queue <lowpriority>.\n",
      "Job <399956> is submitted to default queue <lowpriority>.\n",
      "Job <399957> is submitted to default queue <lowpriority>.\n",
      "Job <399958> is submitted to default queue <lowpriority>.\n",
      "Job <399959> is submitted to default queue <lowpriority>.\n",
      "Job <399960> is submitted to default queue <lowpriority>.\n",
      "Job <399961> is submitted to default queue <lowpriority>.\n",
      "Job <399962> is submitted to default queue <lowpriority>.\n",
      "Job <399963> is submitted to default queue <lowpriority>.\n",
      "Job <399964> is submitted to default queue <lowpriority>.\n",
      "Job <399965> is submitted to default queue <lowpriority>.\n",
      "Job <399966> is submitted to default queue <lowpriority>.\n",
      "Job <399967> is submitted to default queue <lowpriority>.\n",
      "Job <399968> is submitted to default queue <lowpriority>.\n",
      "Job <399969> is submitted to default queue <lowpriority>.\n",
      "Job <399970> is submitted to default queue <lowpriority>.\n",
      "Job <399971> is submitted to default queue <lowpriority>.\n",
      "Job <399972> is submitted to default queue <lowpriority>.\n",
      "Job <399973> is submitted to default queue <lowpriority>.\n",
      "Job <399974> is submitted to default queue <lowpriority>.\n",
      "Job <399975> is submitted to default queue <lowpriority>.\n",
      "Job <399976> is submitted to default queue <lowpriority>.\n",
      "Job <399977> is submitted to default queue <lowpriority>.\n",
      "Job <399978> is submitted to default queue <lowpriority>.\n",
      "Job <399979> is submitted to default queue <lowpriority>.\n",
      "Job <399980> is submitted to default queue <lowpriority>.\n",
      "Job <399981> is submitted to default queue <lowpriority>.\n",
      "Job <399982> is submitted to default queue <lowpriority>.\n",
      "Job <399983> is submitted to default queue <lowpriority>.\n",
      "Job <399984> is submitted to default queue <lowpriority>.\n",
      "Job <399985> is submitted to default queue <lowpriority>.\n",
      "Job <399986> is submitted to default queue <lowpriority>.\n",
      "Job <399987> is submitted to default queue <lowpriority>.\n",
      "Job <399988> is submitted to default queue <lowpriority>.\n",
      "Job <399989> is submitted to default queue <lowpriority>.\n",
      "Job <399990> is submitted to default queue <lowpriority>.\n",
      "Job <399991> is submitted to default queue <lowpriority>.\n",
      "Job <399992> is submitted to default queue <lowpriority>.\n",
      "Job <399993> is submitted to default queue <lowpriority>.\n",
      "Job <399994> is submitted to default queue <lowpriority>.\n",
      "Job <399995> is submitted to default queue <lowpriority>.\n",
      "Job <399996> is submitted to default queue <lowpriority>.\n",
      "Job <399997> is submitted to default queue <lowpriority>.\n",
      "Job <399998> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "module load samtools/1.9\n",
    "bams=$(ls /powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/*.bam)\n",
    "OUT=/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/stats\n",
    "LOG=/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/log\n",
    "for bam in $bams\n",
    "do\n",
    "NAME=`basename ${bam}`\n",
    "PREFIX=`echo ${NAME} | awk -F. '{print $1}'`\n",
    "COMMAND=\"samtools stats ${bam} > ${OUT}/${PREFIX}.stats\"\n",
    "bsub -o ${LOG}/stats.out -e ${LOG}/stats.err -J SAMSTATS ${COMMAND}\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <400015> is submitted to default queue <lowpriority>.\n",
      "Job <400016> is submitted to default queue <lowpriority>.\n",
      "Job <400017> is submitted to default queue <lowpriority>.\n",
      "Job <400018> is submitted to default queue <lowpriority>.\n",
      "Job <400019> is submitted to default queue <lowpriority>.\n",
      "Job <400020> is submitted to default queue <lowpriority>.\n",
      "Job <400021> is submitted to default queue <lowpriority>.\n",
      "Job <400022> is submitted to default queue <lowpriority>.\n",
      "Job <400023> is submitted to default queue <lowpriority>.\n",
      "Job <400024> is submitted to default queue <lowpriority>.\n",
      "Job <400025> is submitted to default queue <lowpriority>.\n",
      "Job <400026> is submitted to default queue <lowpriority>.\n",
      "Job <400027> is submitted to default queue <lowpriority>.\n",
      "Job <400028> is submitted to default queue <lowpriority>.\n",
      "Job <400029> is submitted to default queue <lowpriority>.\n",
      "Job <400030> is submitted to default queue <lowpriority>.\n",
      "Job <400031> is submitted to default queue <lowpriority>.\n",
      "Job <400032> is submitted to default queue <lowpriority>.\n",
      "Job <400033> is submitted to default queue <lowpriority>.\n",
      "Job <400034> is submitted to default queue <lowpriority>.\n",
      "Job <400035> is submitted to default queue <lowpriority>.\n",
      "Job <400036> is submitted to default queue <lowpriority>.\n",
      "Job <400037> is submitted to default queue <lowpriority>.\n",
      "Job <400038> is submitted to default queue <lowpriority>.\n",
      "Job <400039> is submitted to default queue <lowpriority>.\n",
      "Job <400040> is submitted to default queue <lowpriority>.\n",
      "Job <400041> is submitted to default queue <lowpriority>.\n",
      "Job <400042> is submitted to default queue <lowpriority>.\n",
      "Job <400043> is submitted to default queue <lowpriority>.\n",
      "Job <400044> is submitted to default queue <lowpriority>.\n",
      "Job <400045> is submitted to default queue <lowpriority>.\n",
      "Job <400046> is submitted to default queue <lowpriority>.\n",
      "Job <400047> is submitted to default queue <lowpriority>.\n",
      "Job <400048> is submitted to default queue <lowpriority>.\n",
      "Job <400049> is submitted to default queue <lowpriority>.\n",
      "Job <400050> is submitted to default queue <lowpriority>.\n",
      "Job <400051> is submitted to default queue <lowpriority>.\n",
      "Job <400052> is submitted to default queue <lowpriority>.\n",
      "Job <400053> is submitted to default queue <lowpriority>.\n",
      "Job <400054> is submitted to default queue <lowpriority>.\n",
      "Job <400055> is submitted to default queue <lowpriority>.\n",
      "Job <400056> is submitted to default queue <lowpriority>.\n",
      "Job <400057> is submitted to default queue <lowpriority>.\n",
      "Job <400058> is submitted to default queue <lowpriority>.\n",
      "Job <400059> is submitted to default queue <lowpriority>.\n",
      "Job <400060> is submitted to default queue <lowpriority>.\n",
      "Job <400061> is submitted to default queue <lowpriority>.\n",
      "Job <400062> is submitted to default queue <lowpriority>.\n",
      "Job <400063> is submitted to default queue <lowpriority>.\n",
      "Job <400064> is submitted to default queue <lowpriority>.\n",
      "Job <400065> is submitted to default queue <lowpriority>.\n",
      "Job <400066> is submitted to default queue <lowpriority>.\n",
      "Job <400067> is submitted to default queue <lowpriority>.\n",
      "Job <400068> is submitted to default queue <lowpriority>.\n",
      "Job <400069> is submitted to default queue <lowpriority>.\n",
      "Job <400070> is submitted to default queue <lowpriority>.\n",
      "Job <400071> is submitted to default queue <lowpriority>.\n",
      "Job <400072> is submitted to default queue <lowpriority>.\n",
      "Job <400073> is submitted to default queue <lowpriority>.\n",
      "Job <400074> is submitted to default queue <lowpriority>.\n",
      "Job <400075> is submitted to default queue <lowpriority>.\n",
      "Job <400076> is submitted to default queue <lowpriority>.\n",
      "Job <400077> is submitted to default queue <lowpriority>.\n",
      "Job <400078> is submitted to default queue <lowpriority>.\n",
      "Job <400079> is submitted to default queue <lowpriority>.\n",
      "Job <400080> is submitted to default queue <lowpriority>.\n",
      "Job <400081> is submitted to default queue <lowpriority>.\n",
      "Job <400082> is submitted to default queue <lowpriority>.\n",
      "Job <400083> is submitted to default queue <lowpriority>.\n",
      "Job <400084> is submitted to default queue <lowpriority>.\n",
      "Job <400085> is submitted to default queue <lowpriority>.\n",
      "Job <400086> is submitted to default queue <lowpriority>.\n",
      "Job <400087> is submitted to default queue <lowpriority>.\n",
      "Job <400088> is submitted to default queue <lowpriority>.\n",
      "Job <400089> is submitted to default queue <lowpriority>.\n",
      "Job <400090> is submitted to default queue <lowpriority>.\n",
      "Job <400091> is submitted to default queue <lowpriority>.\n",
      "Job <400092> is submitted to default queue <lowpriority>.\n",
      "Job <400093> is submitted to default queue <lowpriority>.\n",
      "Job <400094> is submitted to default queue <lowpriority>.\n",
      "Job <400095> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "STATS=`ls /workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/stats/*`\n",
    "OUT=\"/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/stats/plot\"\n",
    "module load samtools\n",
    "mkdir -p ${OUT}\n",
    "for stats in ${STATS}\n",
    "    do\n",
    "        NAME=`basename ${stats}`\n",
    "        PREFIX=`echo ${NAME} | awk -F. '{print $1}'`\n",
    "        COMMAND=\"plot-bamstats ${stats} -p ${OUT}/${PREFIX}/\"\n",
    "        bsub -J PLOT ${COMMAND}\n",
    "    done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "333.696\n",
      "89.22\n",
      "139\n"
     ]
    }
   ],
   "source": [
    "cat /workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/stats/*stats | egrep \"insert size average\" | awk '{sum+=$NF; c++} END {print sum/c}'\n",
    "cat /workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/stats/*stats | egrep \"insert size standard deviation\" | awk '{sum+=$NF; c++} END {print sum/c}'\n",
    "cat /workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/stats/*stats | egrep \"average length\" | awk '{sum+=$NF; c++} END {print sum/c}'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "---------- /software/OSutils/modules-4.4.1/share/Modules/modulefiles -----------\n",
      "1SynChro/1.15\n",
      "3d-dna/180922\n",
      "A5-miseq/20160825\n",
      "abyss/2.0.2\n",
      "abyss/2.1.0\n",
      "abyss/2.1.1\n",
      "abyss/2.1.2\n",
      "abyss/2.1.3\n",
      "abyss/2.1.4\n",
      "abyss/2.1.5\n",
      "adapterremoval/2.2.2\n",
      "AfterQC/0.9.1\n",
      "AfterQC/0.9.5\n",
      "allpathslg/44837\n",
      "allpathslg/44913\n",
      "allpathslg/49722\n",
      "allpathslg/50191\n",
      "allpathslg/52488\n",
      "amos/3.1.0\n",
      "angsd/0.913\n",
      "angsd/0.917\n",
      "angsd/0.917-116-g5d087b2\n",
      "angsd/0.918\n",
      "angsd/0.919\n",
      "angsd/0.921\n",
      "angsd/0.922\n",
      "angsd/0.923\n",
      "angsd/0.925\n",
      "angsd-wrapper/7224db2\n",
      "anytag/2.5.2\n",
      "apsim/7.10-r49ace54f9c8a670190aef9d8d0fb9d5477bb1534\n",
      "apsim/7.9-r4047\n",
      "apsimx/2018.01.30.2253\n",
      "apsimx/2018.09.28.3099\n",
      "apsimx/2019.01.08.3392\n",
      "apsimx/2019.04.03.3693\n",
      "apsimx/2019.06.05.3920\n",
      "apsimx/2019.07.18.4025\n",
      "apsimx/2019.10.04.4236\n",
      "apsimx/2020.04.09.5012\n",
      "apsimx/2020.08.04.5350\n",
      "arcs/1.0.6\n",
      "art/MountRainier\n",
      "aspera-cli/3.7.2.354.010c3b8\n",
      "assemblathon_stats/14dfdab\n",
      "assemblathon_stats/2011_10_13\n",
      "asub/2.1\n",
      "augustus/3.2.2\n",
      "augustus/3.3\n",
      "augustus/3.3.1\n",
      "augustus/3.3.3-new\n",
      "aws-cli/2.0.43\n",
      "bam2fastx/1.3.0\n",
      "bambam/1.4\n",
      "bamtools/2.3.0\n",
      "bamtools/2.4.0\n",
      "bats/0.4.0\n",
      "BBMap/37.28\n",
      "BBMap/37.53\n",
      "BBMap/37.93\n",
      "BBMap/38.33\n",
      "bcftools/1.10.2\n",
      "bcftools/1.3.1\n",
      "bcftools/1.6\n",
      "bcftools/1.9\n",
      "bcgTree/1.0.8\n",
      "beagle/4.0\n",
      "beagle/4.1\n",
      "BEAST/1.8.4\n",
      "BEAST2/2.4.7\n",
      "bedops/2.4.39\n",
      "bedtools/2.20.1\n",
      "bedtools/2.21.0\n",
      "bedtools/2.27.1\n",
      "bfc-submitted/v1\n",
      "biobambam/0.0.191\n",
      "biobambam2/2.0.106\n",
      "biobambam2/2.0.44\n",
      "bismark/0.19.1\n",
      "bismark/0.20.0\n",
      "blasr/5.2.badc092\n",
      "blast/2.2.26\n",
      "BLAT/36\n",
      "blobtools/1.0\n",
      "boost/1_61_0\n",
      "boost/1_61_0-dev\n",
      "boost/1_65_1\n",
      "boost/1_65_1-dev\n",
      "bowtie/1.0.0\n",
      "bowtie2/2.2.5\n",
      "bowtie2/2.2.9\n",
      "bowtie2/2.3.4.3\n",
      "BRAKER/1.11\n",
      "BRAKER/1.9\n",
      "BRAKER/2.0.1\n",
      "BRAKER/2.0.4\n",
      "BRAKER/2.0.5\n",
      "BRAKER/2.1.0\n",
      "BRAKER/2.1.2\n",
      "BRAKER/2.1.5\n",
      "breakdancer/4e44b43\n",
      "BSseeker2/2.1.5\n",
      "BUSCO/v1.2\n",
      "BUSCO/v2.0\n",
      "BUSCO/v3.0.2\n",
      "BUSCO/v4.1.2\n",
      "bwa/0.7.12\n",
      "bwa/0.7.15\n",
      "bwa/0.7.17\n",
      "canu/1.3\n",
      "canu/1.5\n",
      "canu/1.6\n",
      "canu/1.7\n",
      "cd-hit/4.6.1\n",
      "cdo/1.7.0\n",
      "cdo/1.9.3\n",
      "cdo/1.9.5\n",
      "cegma/v2.4.010312\n",
      "cegma/v2.5\n",
      "checkm/1.0.10\n",
      "checkm/1.0.11\n",
      "checkm/1.0.12\n",
      "checkm/1.0.13\n",
      "checkm/1.0.6\n",
      "checkm/1.0.7\n",
      "checkm/1.0.8\n",
      "checkm/1.1.3\n",
      "chromonomer/1.07\n",
      "chromosomer/0.1.3\n",
      "cndsrc/2010.10.11\n",
      "cndsrc/2017.11.30\n",
      "cnvnator/0.3.3\n",
      "cnvnator/0.4.1\n",
      "CodingQuarry/2.0\n",
      "COGNATE/v1.0\n",
      "colordiff/1.0.18\n",
      "conda/4.5.1\n",
      "conda/4.8.2\n",
      "cookiecutter/1.0.0\n",
      "crema/173e9a4\n",
      "crisp/c6cd86e\n",
      "crossmap/0.3.3\n",
      "csvkit/1.0.2\n",
      "cufflinks/2.2.1\n",
      "cutadapt/1.10\n",
      "cutadapt/1.15\n",
      "dammit/1.2\n",
      "data-registry/1.0\n",
      "data-registry/1.0-dev\n",
      "data-registry/1.0-test\n",
      "deepvariant/0.7.2-py27h5d9141f_1\n",
      "diamond/0.9.24\n",
      "Disco/1.1-beta\n",
      "docker-compose/1.11.2\n",
      "docker-compose/1.7.1\n",
      "dtool/3.25.0\n",
      "dvc/1.6.1\n",
      "DWGSIM/0561546\n",
      "ea-utils/1.04.807\n",
      "ea-utils/1.1.2-484\n",
      "ea-utils/1.1.2-537\n",
      "ea-utils/1.1.2-806\n",
      "eddypro-engine/5.1.1\n",
      "eddypro-engine/5.2.0\n",
      "eddypro-engine/5.2.1\n",
      "eddypro-engine/6.0.0\n",
      "eddypro-engine/6.1.0\n",
      "eddypro-engine/6.2.0\n",
      "eddypro-engine/6.2.1\n",
      "edta/1.8.3\n",
      "EffectorP/1.0\n",
      "eg-all/91\n",
      "eg-web-bacteria/91\n",
      "eg-web-common/91\n",
      "eg-web-fungi/91\n",
      "eg-web-metazoa/91\n",
      "eg-web-plants/91\n",
      "eg-web-protists/91\n",
      "eg-web-search/91\n",
      "eg-web-testsuite/91\n",
      "ehive/2.4\n",
      "eigensoft/7.2.1\n",
      "eman2/2.3\n",
      "emboss/6.4.0\n",
      "emboss/6.5.7\n",
      "emboss/6.6.0\n",
      "emirge/0.61.1\n",
      "ensembl/69\n",
      "ensembl/91\n",
      "ensembl/92\n",
      "ensembl/93\n",
      "ensembl-analysis/69\n",
      "ensembl-analysis/91\n",
      "ensembl-analysis/92\n",
      "ensembl-analysis/93\n",
      "ensembl-api/69\n",
      "ensembl-api/91\n",
      "ensembl-api/92\n",
      "ensembl-api/93\n",
      "ensembl-compara/69\n",
      "ensembl-compara/91\n",
      "ensembl-compara/92\n",
      "ensembl-compara/93\n",
      "ensembl-funcgen/69\n",
      "ensembl-funcgen/91\n",
      "ensembl-funcgen/92\n",
      "ensembl-funcgen/93\n",
      "ensembl-genebuild/69\n",
      "ensembl-genebuild/91\n",
      "ensembl-genebuild/92\n",
      "ensembl-genebuild/93\n",
      "ensemblgenomes-api/91\n",
      "ensembl-git-tools/d55068a\n",
      "ensembl-hive/69\n",
      "ensembl-hive/91\n",
      "ensembl-hive/92\n",
      "ensembl-hive/93\n",
      "ensembl-hive-openlava/69\n",
      "ensembl-hive-openlava/91\n",
      "ensembl-hive-openlava/92\n",
      "ensembl-hive-openlava/93\n",
      "ensembl-io/69\n",
      "ensembl-io/91\n",
      "ensembl-io/92\n",
      "ensembl-io/93\n",
      "ensembl-killlist/69\n",
      "ensembl-killlist/91\n",
      "ensembl-killlist/92\n",
      "ensembl-killlist/93\n",
      "ensembl-metadata/69\n",
      "ensembl-metadata/91\n",
      "ensembl-metadata/92\n",
      "ensembl-metadata/93\n",
      "ensembl-notebook/latest\n",
      "ensembl-orm/69\n",
      "ensembl-orm/91\n",
      "ensembl-orm/92\n",
      "ensembl-orm/93\n",
      "ensembl-pipeline/69\n",
      "ensembl-pipeline/91\n",
      "ensembl-pipeline/92\n",
      "ensembl-pipeline/93\n",
      "ensembl-production/69\n",
      "ensembl-production/91\n",
      "ensembl-production/92\n",
      "ensembl-production/93\n",
      "ensembl-taxonomy/69\n",
      "ensembl-taxonomy/91\n",
      "ensembl-taxonomy/92\n",
      "ensembl-taxonomy/93\n",
      "ensembl-tools/69\n",
      "ensembl-tools/91\n",
      "ensembl-tools/92\n",
      "ensembl-tools/93\n",
      "ensembl-variation/69\n",
      "ensembl-variation/91\n",
      "ensembl-variation/92\n",
      "ensembl-variation/93\n",
      "ensembl-vep/69\n",
      "ensembl-vep/91\n",
      "ensembl-vep/92\n",
      "ensembl-vep/93\n",
      "entrez-direct/13.8.20200819\n",
      "e-PCR/2.3.9\n",
      "ericscript/0.5.5\n",
      "e-utilities/5.40\n",
      "EVidenceModeler/1.1.1\n",
      "evigene/02-11-2017\n",
      "evigene/18-01-2018\n",
      "evigene/18-09-2013\n",
      "exonerate/2.4.0\n",
      "FALCON/1.7.4\n",
      "fasta/35.4.12\n",
      "fasta/36.3.8g\n",
      "fast-gbs/2016.12.14\n",
      "fast-gbs/2017.01.12\n",
      "fastp/0.12.5\n",
      "fastp/0.19.4\n",
      "fastp/0.20.0\n",
      "FastQC/0.11.1\n",
      "FastQC/0.11.2\n",
      "FastQC/0.11.7\n",
      "fastq_screen/v0.11.1\n",
      "fastq_screen/v0.14.0\n",
      "fastq_screen/v0.4.4\n",
      "fastq_screen/v0.5.2\n",
      "faststructure/1.0\n",
      "FastTree/2.1.10\n",
      "fastx_toolkit/0.0.13\n",
      "FGMP/1.0\n",
      "FLASH/1.2.11\n",
      "flye/2.4.2\n",
      "flye/2.5\n",
      "flye/2.7.1\n",
      "freebayes/1.1.0\n",
      "freebayes/1.3.1\n",
      "freebayes/v1.0.2-16-gd466dde\n",
      "freetype/2.9\n",
      "GapFiller/v1.10\n",
      "GARM/0.7.4\n",
      "gatk/1.0\n",
      "gatk/3.7\n",
      "gatk/3.8.0\n",
      "Gblocks/0.91b\n",
      "gdal/2.4.0\n",
      "GEM/20130406-045632\n",
      "gemoma/1.4.2\n",
      "gemoma/1.6.4\n",
      "geneid/v1.4.4\n",
      "GeneMark/4.21\n",
      "GeneMark/4.33\n",
      "GeneMark/4.56\n",
      "genomethreader/1.7.0\n",
      "genometools/1.5.10\n",
      "getorganelle/1.6.2e\n",
      "gffread/0.9.6\n",
      "git/2.11.0\n",
      "git/2.12.2\n",
      "git/2.14.1\n",
      "git/2.21.0\n",
      "git-annex/6.20160613-g35dbe35\n",
      "gitflow/15aab26\n",
      "git-lfs/2.3.0\n",
      "globus-cli/1.2.2\n",
      "globus-cli/1.6.0\n",
      "gmap/2016-06-09\n",
      "gmap/2017-06-20\n",
      "gmap/2017-09-30\n",
      "gmap/2018-03-01\n",
      "gmap/2018-03-25\n",
      "gmap/2018-05-11\n",
      "gmap/2018-05-30\n",
      "gmato/v1.2\n",
      "GMcloser/1.6.2\n",
      "golang/1.14.2\n",
      "golang/1.9.7\n",
      "GroIMP/1.5\n",
      "gromacs/2018\n",
      "gromacs/2018.1\n",
      "gromacs/2018.2\n",
      "gromacs/2018.3\n",
      "gromacs/2018.4\n",
      "gromacs/2018.5\n",
      "gromacs/2019\n",
      "guppy/3.2.4\n",
      "guppy/3.4.4\n",
      "guppy/3.5.2\n",
      "guppy/3.6.1\n",
      "hagfish/05d08d1\n",
      "hagfish/1_May_2013\n",
      "hapcol/97d4a5e\n",
      "hapcompass/0.8.2\n",
      "HapCUT2/0121162\n",
      "Haplosim/ae652a2\n",
      "Haplosim/bd39c6f\n",
      "HapTree/1.0\n",
      "hdf5/1.8.17\n",
      "HiC-pro/2.11.1\n",
      "hicup/0.5.9\n",
      "hicup/0.7.2\n",
      "hisat2/2.1.0\n",
      "hmmer/3.0\n",
      "hmmer/3.1b1\n",
      "hmmer/3.1b2\n",
      "hmmer/3.3\n",
      "HTSeq/0.6.1\n",
      "HTSeq/0.7.2\n",
      "HTSeq/0.9.1\n",
      "htslib/1.10.2\n",
      "htslib/1.3.1\n",
      "htslib/1.7\n",
      "htslib/1.9\n",
      "hub/2.8.4\n",
      "humann2/0.11.1\n",
      "iCommands/4.1.8\n",
      "idba/a1bafe6\n",
      "IGV/2.3.93\n",
      "infernal/1.1.2\n",
      "iqtree/1.6.11\n",
      "JAGS/4.0.0\n",
      "JAGS/4.2.0\n",
      "java/1.7.0\n",
      "java/1.8.0\n",
      "jcvi/0.7.3\n",
      "jcvi/0.7.7\n",
      "jcvi/0.8.12\n",
      "jellyfish/1.1.10\n",
      "jellyfish/2.1.3\n",
      "jellyfish/2.2.10\n",
      "jmodeltest/2.1.10\n",
      "jq/1.6\n",
      "kat/2.3.1\n",
      "kat/2.4.1\n",
      "kentUtils/302.1\n",
      "khmer/2.0\n",
      "KMA/1.2.4\n",
      "KMC/3.0.0\n",
      "kneaddata/0.6.1\n",
      "kraken/0.10.5\n",
      "kraken/1.0\n",
      "kraken/1.1\n",
      "Krona/2.7\n",
      "LACHESIS/20150929\n",
      "last/770\n",
      "lastz/1.02.00\n",
      "LDhat/81596e2\n",
      "links/v1.8.5\n",
      "links/v1.8.6\n",
      "LocHap/2.0\n",
      "luarocks/2.4.1\n",
      "lumpy-sv/0.2.13\n",
      "MACS2/2.1.1\n",
      "mafft/7.307\n",
      "mandalorion-episode-ii/6219d58\n",
      "mapcomp/1.0\n",
      "mapgd/d3edee2\n",
      "MAPGD/0.4.26\n",
      "MAPGD/0.5\n",
      "mason/0.1.2\n",
      "MaSuRCA/3.2.2\n",
      "MaSuRCA/3.2.3\n",
      "MaSuRCA/3.2.4\n",
      "MaSuRCA/3.2.8\n",
      "MaSuRCA/3.3.3\n",
      "mauve/2015.02.13\n",
      "mcl/14-137\n",
      "mcscan/0.8\n",
      "MECAT/3898797\n",
      "megahit/1.1.1\n",
      "megahit/1.1.3\n",
      "megahit/1.2.6\n",
      "megahit/1.2.9\n",
      "meme/4.11.2\n",
      "Meraculous/2.2.4\n",
      "Metaassembler/1.5\n",
      "metabat/0.32.4\n",
      "metabat2/2.15\n",
      "metabat2/2.15-3-g367a7ef\n",
      "metaphlan2/2.6.0\n",
      "microbiome_helper/4330e63\n",
      "mikado/1.1\n",
      "miniasm/0.2\n",
      "miniasm/55cf018\n",
      "miniconda2/4.3.13\n",
      "minimap2/2.17\n",
      "minimap2/2.9\n",
      "mira/4.0.2\n",
      "miRanda/3.3a\n",
      "mitoz/2.3\n",
      "mplus/0.1\n",
      "mrbayes/3.2.7\n",
      "mrbayes/3.2.7a\n",
      "mrbayes/3.2.7a-gpu\n",
      "msms/3.2rc_163\n",
      "multiqc/1.0\n",
      "multiqc/1.2\n",
      "multiqc/1.3\n",
      "multiqc/1.5\n",
      "multiqc/1.6\n",
      "multiqc/1.7\n",
      "multiqc/1.8\n",
      "multiqc/1.9\n",
      "MUMmer/3.23\n",
      "muscle/3.8.1551\n",
      "muscle/4.0.170\n",
      "nanopack/1.0.0\n",
      "ncbi-blast/2.10.1\n",
      "ncbi-blast/2.2.25\n",
      "ncbi-blast/2.2.30\n",
      "ncbi-blast/2.6.0\n",
      "ncbi-genome-download/0.2.12\n",
      "NetLogo/6.0.4\n",
      "newbler/2.5.3\n",
      "newbler/2.7\n",
      "newbler/2.9\n",
      "nextclip/v1.3.1\n",
      "nextgenmap/v0.5.5\n",
      "NextGenMap/v0.5.5\n",
      "ngmlr/0.2.7\n",
      "NovoGraph/1.0.0\n",
      "NOVOPlasty/2.7.2\n",
      "NPStat/v1\n",
      "nQuire/a990a88\n",
      "NSEG/2018-05-30\n",
      "NxTrim/0.4.2\n",
      "NxTrim/v0.4.1-0f17575\n",
      "oases/0.2.09\n",
      "obitools/1.2.11\n",
      "octave/4.2.2\n",
      "octave/4.4.1\n",
      "openbugs/3.2.3\n",
      "opendronemap/0.4.0\n",
      "opendronemap/0.4.1\n",
      "openlava/3.2\n",
      "openmpi/1.8.8\n",
      "openmpi/4.0.2\n",
      "OPERA-LG/v2.0.6\n",
      "orthofinder/2.3.1\n",
      "orthomcl/2.0.9\n",
      "PAML/4.9i\n",
      "pandaseq/2.10\n",
      "pandoc/1.19.2\n",
      "PASA/2.0.2\n",
      "PASA/2.1.0\n",
      "PASA/2.2.0\n",
      "PASA/2.3.3\n",
      "patman/1.2.2\n",
      "pblat/2.0\n",
      "pblat/2.1\n",
      "PBSIM/308622d\n",
      "PBSuite/15.8.24\n",
      "PBSuite/17.05.09-dbrowneup\n",
      "pcl/1.8.1\n",
      "pcl/1.9.0\n",
      "pcl/1.9.1\n",
      "PedigreeSim/v2.0\n",
      "perl/5.10.1\n",
      "perl/5.10.1-thread-multi\n",
      "perl/5.12.5\n",
      "perl/5.12.5-thread-multi\n",
      "perl/5.14.4\n",
      "perl/5.14.4-thread-multi\n",
      "perl/5.16.3\n",
      "perl/5.16.3-thread-multi\n",
      "perl/5.18.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl/5.20.3\n",
      "perl/5.20.3-thread-multi\n",
      "perl/5.22.2\n",
      "perl/5.22.2-thread-multi\n",
      "perl/5.24.0\n",
      "perl/5.24.0-thread-multi\n",
      "perl/5.24.1\n",
      "perl/5.24.1-thread-multi\n",
      "perl/5.26.0\n",
      "perl/5.26.0-thread-multi\n",
      "perl/5.28.0\n",
      "perl/5.28.0-thread-multi\n",
      "perl/5.30.2\n",
      "perl/5.30.2-thread-multi\n",
      "perl/5.8.9\n",
      "perl/5.8.9-thread-multi\n",
      "perl/bio-extras\n",
      "perl/bio-extras-5.24.0-legacy\n",
      "perl/core\n",
      "perl/ensembl\n",
      "perl/jupyterhub\n",
      "perlbrew/0.76\n",
      "pfr-python2/2.7.13\n",
      "pfr-python3/3.6.1\n",
      "pfr-python3/3.6.5\n",
      "pfr-python3/3.6.6\n",
      "phylip/3.696\n",
      "phylobayes/4.1c\n",
      "PhyloSift/1.0.1\n",
      "picard-tools/1.112\n",
      "picard-tools/1.79\n",
      "picard-tools/2.10.1\n",
      "picard-tools/2.18.7\n",
      "picard-tools/2.2.4\n",
      "picard-tools/2.9.4\n",
      "picrust2/2.3.0\n",
      "pilon/1.18\n",
      "pilon/1.20\n",
      "pilon/1.23\n",
      "pinfish/0.1.0\n",
      "platanus/1.2.4\n",
      "platanus_trim/1.0.7\n",
      "Platypus/0.8.1\n",
      "plink/1.07\n",
      "plink/1.90b6.5\n",
      "plink2/v2.00a2lm\n",
      "plncpro/1.1\n",
      "popoolation2/1201\n",
      "porechop/0.2.3\n",
      "portcullis/1.1.0\n",
      "portcullis/1.1.1\n",
      "portcullis/1.1.2\n",
      "powerPlant/core\n",
      "powerPlant/locate\n",
      "pplacer/v1.1.alpha17-6-g5cecf99\n",
      "ppsPCP/1.0\n",
      "primer3/2.2.3\n",
      "primer3/2.3.5\n",
      "prinseq-lite/0.20.4\n",
      "Prodigal/2.6.3\n",
      "proj/5.2.0\n",
      "prokka/1.11\n",
      "prokka/1.13\n",
      "proteinortho/6.0.15\n",
      "purge_haplotigs/1.1.0\n",
      "pyarrow/0.16.0\n",
      "pycoqc/2.5.0.21\n",
      "qiime/1.9.1\n",
      "qiime2/2018.11\n",
      "qiime2/2018.2\n",
      "qiime2/2019.1\n",
      "qiime2/2019.10\n",
      "qiime2/2019.1-picrust2\n",
      "qiime2/2019.4\n",
      "qiime2/2019.7\n",
      "qiime2/2019.7-picrust2\n",
      "qiime2/2020.2\n",
      "qiime2/2020.6\n",
      "qualimap/v2.2.1\n",
      "quast/4.2\n",
      "R/3.3.0\n",
      "R/3.3.1\n",
      "R/3.3.3\n",
      "R/3.4.0\n",
      "R/3.4.1\n",
      "R/3.4.2\n",
      "R/3.4.3\n",
      "R/3.4.4\n",
      "R/3.5.0\n",
      "R/3.5.2\n",
      "R/3.5.3\n",
      "R/3.6.1\n",
      "R/3.6.2\n",
      "R/3.6.3\n",
      "R/4.0.0\n",
      "racon/1.3.0\n",
      "racon/1.3.1\n",
      "racon/1.3.2\n",
      "racon/1.3.3\n",
      "racon/1.4.0\n",
      "racon/1.4.2\n",
      "racon/1.4.3\n",
      "racon/1.4.7\n",
      "ragoo/1.01\n",
      "ragoo/1.02\n",
      "RAILS/v1.2\n",
      "RANBOW/1\n",
      "RANBOW/1-temp\n",
      "RECON/1.08\n",
      "REPCLASS/1.0.1\n",
      "RepeatMasker-open/3-3-0\n",
      "RepeatMasker-open/4-0-3\n",
      "RepeatMasker-open/4-0-5\n",
      "RepeatModeler/1.0.8\n",
      "RepeatModeler-open/1.0.11\n",
      "RepeatScout/1.0.5\n",
      "REPET/2.5\n",
      "rnacocktail/0.2.1\n",
      "RSEM/1.3.0\n",
      "rtg-tools/3.7.1-eb13bbb\n",
      "ruby/2.1.9\n",
      "ruby/2.2.7\n",
      "ruby/2.3.4\n",
      "ruby/2.4.1\n",
      "sabre/1.000\n",
      "salmon/1.1.0\n",
      "salmon/1.3.0\n",
      "salsa/v2.2\n",
      "samtools/0.1.18\n",
      "samtools/0.1.19\n",
      "samtools/1.1\n",
      "samtools/1.10\n",
      "samtools/1.2\n",
      "samtools/1.3.1\n",
      "samtools/1.7\n",
      "samtools/1.9\n",
      "ScanIndel/1.2dev\n",
      "SDhap/56d7678\n",
      "seqkit/0.7.0\n",
      "seqmagick/0.8.0\n",
      "seqtk/1.2\n",
      "sex-detector-plusplus/00f7d723\n",
      "sga/0.10.10-2\n",
      "sga/0.10.13\n",
      "sga/0.10.15\n",
      "sga/0.9.19\n",
      "sga/0.9.35\n",
      "SGSGeneLoss/0.1\n",
      "singularity/2.4.2\n",
      "singularity/2.5.1\n",
      "singularity/3\n",
      "SLiM/3.1\n",
      "smalt/0.7.6\n",
      "snakemake/5.10.0\n",
      "snap/1.0beta.18\n",
      "snapr/0.16alpha.51\n",
      "sniffles/1.0.11\n",
      "SnpEff/4.3t\n",
      "SNPhylo/20140701\n",
      "SNPhylo/20160204\n",
      "SOAPdenovo/1.05\n",
      "SOAPdenovo/2.04\n",
      "SOAPdenovo-Trans/1.03\n",
      "SolexaQA/3.1.5\n",
      "sortmerna/2.1\n",
      "sortmerna/3.0.3\n",
      "sortmerna/4.2.0\n",
      "SPAdes/3.10.1\n",
      "SPAdes/3.11.1\n",
      "SPAdes/3.12.0\n",
      "SPAdes/3.13.0\n",
      "SPAdes/3.14.0\n",
      "SPAdes/3.7.1\n",
      "SqueezeMeta/0.4.4\n",
      "SqueezeMeta/1.0.0-beta\n",
      "sratoolkit/2.8.2.1\n",
      "SSPACE/2.0\n",
      "stacks/1.40\n",
      "stacks/1.40_withbam\n",
      "stacks/2.0\n",
      "stacks/2.1\n",
      "stacks/2.2\n",
      "stamp/2.1.3\n",
      "standard-RAxML/8.2.11\n",
      "standard-RAxML/8.2.9\n",
      "STAR/2.4.1d\n",
      "STAR/2.4.2a\n",
      "STAR/2.5.2a\n",
      "STAR/2.5.2b\n",
      "STAR/2.5.3a\n",
      "STAR/2.6.1a\n",
      "STAR/2.6.1d\n",
      "STAR-Fusion/0.8.0\n",
      "stringtie/1.3.3b.Linux_x86_64\n",
      "subread/1.4.5\n",
      "subread/1.5.3\n",
      "supernova/1.2.2\n",
      "survivor/1.0.5\n",
      "SV2/1.4.3.2\n",
      "svaba/1.1.3\n",
      "svim/1.2.0\n",
      "svtools/0.3.2\n",
      "swan/3516c2f\n",
      "tabix/0.2.6\n",
      "TargQC/1.4.4\n",
      "tasr/1.6.2\n",
      "tassel/3.0\n",
      "tassel/4.0\n",
      "tassel/5.0\n",
      "tassel/5.2.28\n",
      "taxmaps/0.2.1\n",
      "tetools/1.1\n",
      "texlive/20151117\n",
      "tigmint/1.1.2\n",
      "tomahawk/0.7.1\n",
      "tophat/2.0.13\n",
      "tophat/2.1.1\n",
      "totalview/2017.2.11\n",
      "TransDecoder/3.0.1\n",
      "TransDecoder/5.0.1\n",
      "TransDecoder/5.5.0\n",
      "transrate/1.0.3\n",
      "trf/4.07\n",
      "trim_galore/0.4.1\n",
      "trim_galore/0.4.3\n",
      "Trimmomatic/0.36\n",
      "trinityrnaseq/2.0.6\n",
      "trinityrnaseq/2.10.0\n",
      "trinityrnaseq/2.2.0\n",
      "trinityrnaseq/2.3.2\n",
      "trinityrnaseq/2.4.0\n",
      "trinityrnaseq/2.5.1\n",
      "trinityrnaseq/2.6.5\n",
      "trinityrnaseq/2.8.6\n",
      "trinityrnaseq/2.9.0\n",
      "trinityrnaseq/2.9.1\n",
      "trinotate/3.0.2\n",
      "trinotate/3.2.0\n",
      "tRNAscan-SE/1.3.1\n",
      "tRNAscan-SE/2.0\n",
      "ucsc-pslcdnafilter/324\n",
      "unicycler/0.4.8\n",
      "unixODBC/2.3.0\n",
      "vcflib/50a3229\n",
      "vcflib/586c5ae\n",
      "vcflib/73b7f13\n",
      "vcftools/0.1.12\n",
      "vcftools/0.1.14\n",
      "velvet/1.2.10\n",
      "VersioningService/69\n",
      "VersioningService/91\n",
      "VersioningService/92\n",
      "VersioningService/93\n",
      "vg/1.10.0\n",
      "vg/1.11.0\n",
      "vg/1.12.0\n",
      "vg/1.12.1\n",
      "vg/1.13.0\n",
      "vg/1.7.0\n",
      "vg/1.8.0\n",
      "vg/1.9.0\n",
      "ViennaRNA/2.2.2\n",
      "vsearch/2.3.2\n",
      "wgs/8.3rc2\n",
      "whatshap/0.14.1-py36_0\n",
      "wise/2.4.1\n",
      "wkhtmlox/0.12.3\n",
      "wtdbg2/2.2\n"
     ]
    }
   ],
   "source": [
    "module avail"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <400703> is submitted to default queue <lowpriority>.\n",
      "Job <400704> is submitted to default queue <lowpriority>.\n",
      "Job <400705> is submitted to default queue <lowpriority>.\n",
      "Job <400706> is submitted to default queue <lowpriority>.\n",
      "Job <400707> is submitted to default queue <lowpriority>.\n",
      "Job <400708> is submitted to default queue <lowpriority>.\n",
      "Job <400709> is submitted to default queue <lowpriority>.\n",
      "Job <400710> is submitted to default queue <lowpriority>.\n",
      "Job <400711> is submitted to default queue <lowpriority>.\n",
      "Job <400712> is submitted to default queue <lowpriority>.\n",
      "Job <400713> is submitted to default queue <lowpriority>.\n",
      "Job <400714> is submitted to default queue <lowpriority>.\n",
      "Job <400715> is submitted to default queue <lowpriority>.\n",
      "Job <400716> is submitted to default queue <lowpriority>.\n",
      "Job <400717> is submitted to default queue <lowpriority>.\n",
      "Job <400718> is submitted to default queue <lowpriority>.\n",
      "Job <400719> is submitted to default queue <lowpriority>.\n",
      "Job <400720> is submitted to default queue <lowpriority>.\n",
      "Job <400721> is submitted to default queue <lowpriority>.\n",
      "Job <400722> is submitted to default queue <lowpriority>.\n",
      "Job <400723> is submitted to default queue <lowpriority>.\n",
      "Job <400724> is submitted to default queue <lowpriority>.\n",
      "Job <400725> is submitted to default queue <lowpriority>.\n",
      "Job <400726> is submitted to default queue <lowpriority>.\n",
      "Job <400727> is submitted to default queue <lowpriority>.\n",
      "Job <400728> is submitted to default queue <lowpriority>.\n",
      "Job <400729> is submitted to default queue <lowpriority>.\n",
      "Job <400730> is submitted to default queue <lowpriority>.\n",
      "Job <400731> is submitted to default queue <lowpriority>.\n",
      "Job <400732> is submitted to default queue <lowpriority>.\n",
      "Job <400733> is submitted to default queue <lowpriority>.\n",
      "Job <400734> is submitted to default queue <lowpriority>.\n",
      "Job <400735> is submitted to default queue <lowpriority>.\n",
      "Job <400736> is submitted to default queue <lowpriority>.\n",
      "Job <400737> is submitted to default queue <lowpriority>.\n",
      "Job <400738> is submitted to default queue <lowpriority>.\n",
      "Job <400739> is submitted to default queue <lowpriority>.\n",
      "Job <400740> is submitted to default queue <lowpriority>.\n",
      "Job <400741> is submitted to default queue <lowpriority>.\n",
      "Job <400742> is submitted to default queue <lowpriority>.\n",
      "Job <400743> is submitted to default queue <lowpriority>.\n",
      "Job <400744> is submitted to default queue <lowpriority>.\n",
      "Job <400745> is submitted to default queue <lowpriority>.\n",
      "Job <400746> is submitted to default queue <lowpriority>.\n",
      "Job <400747> is submitted to default queue <lowpriority>.\n",
      "Job <400748> is submitted to default queue <lowpriority>.\n",
      "Job <400749> is submitted to default queue <lowpriority>.\n",
      "Job <400750> is submitted to default queue <lowpriority>.\n",
      "Job <400751> is submitted to default queue <lowpriority>.\n",
      "Job <400752> is submitted to default queue <lowpriority>.\n",
      "Job <400753> is submitted to default queue <lowpriority>.\n",
      "Job <400754> is submitted to default queue <lowpriority>.\n",
      "Job <400755> is submitted to default queue <lowpriority>.\n",
      "Job <400756> is submitted to default queue <lowpriority>.\n",
      "Job <400757> is submitted to default queue <lowpriority>.\n",
      "Job <400758> is submitted to default queue <lowpriority>.\n",
      "Job <400759> is submitted to default queue <lowpriority>.\n",
      "Job <400760> is submitted to default queue <lowpriority>.\n",
      "Job <400761> is submitted to default queue <lowpriority>.\n",
      "Job <400762> is submitted to default queue <lowpriority>.\n",
      "Job <400763> is submitted to default queue <lowpriority>.\n",
      "Job <400764> is submitted to default queue <lowpriority>.\n",
      "Job <400765> is submitted to default queue <lowpriority>.\n",
      "Job <400766> is submitted to default queue <lowpriority>.\n",
      "Job <400767> is submitted to default queue <lowpriority>.\n",
      "Job <400768> is submitted to default queue <lowpriority>.\n",
      "Job <400769> is submitted to default queue <lowpriority>.\n",
      "Job <400770> is submitted to default queue <lowpriority>.\n",
      "Job <400771> is submitted to default queue <lowpriority>.\n",
      "Job <400772> is submitted to default queue <lowpriority>.\n",
      "Job <400773> is submitted to default queue <lowpriority>.\n",
      "Job <400774> is submitted to default queue <lowpriority>.\n",
      "Job <400775> is submitted to default queue <lowpriority>.\n",
      "Job <400776> is submitted to default queue <lowpriority>.\n",
      "Job <400777> is submitted to default queue <lowpriority>.\n",
      "Job <400778> is submitted to default queue <lowpriority>.\n",
      "Job <400779> is submitted to default queue <lowpriority>.\n",
      "Job <400780> is submitted to default queue <lowpriority>.\n",
      "Job <400781> is submitted to default queue <lowpriority>.\n",
      "Job <400782> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "module load gatk/3.8.0\n",
    "bams=$(ls /powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/*.bam)\n",
    "OUT=/workspace/hramzr/2_Phd_PROJECT/WGS/RTC\n",
    "LOG=\"${OUT}/logs\"\n",
    "GATK=\"/powerplant/workspace/cflcyd/software/GATK/gatk-3.8.1/GenomeAnalysisTK.jar\"\n",
    "JAVA=\"java -jar -Xmx64G\"\n",
    "ref=\"/workspace/hramzr/2_Phd_PROJECT/snapper_genome/Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta\"\n",
    "for bam in ${bams}\n",
    "    do\n",
    "        NAME=`basename ${bam}`\n",
    "        PREFIX=`echo ${NAME} | awk -F[_] '{print $1}'`\n",
    "        COMMAND=\"${JAVA} ${GATK} -T RealignerTargetCreator \\\n",
    "                    -R ${ref} \\\n",
    "                    -I ${bam} \\\n",
    "                    -maxInterval 420 \\\n",
    "                    -o ${OUT}/${PREFIX}.intervals\"\n",
    "                    \n",
    "        bsub -o ${LOG}/${PREFIX}.out -e ${LOG}/${PREFIX}.err \\\n",
    "             -J \"RTC ${PREFIX}\" \\\n",
    "             -n 3 \\\n",
    "             -R \"rusage[mem=36000] span[hosts=1]\" \\\n",
    "             ${COMMAND}\n",
    "    done\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <400804> is submitted to default queue <lowpriority>.\n",
      "Job <400805> is submitted to default queue <lowpriority>.\n",
      "Job <400806> is submitted to default queue <lowpriority>.\n",
      "Job <400807> is submitted to default queue <lowpriority>.\n",
      "Job <400808> is submitted to default queue <lowpriority>.\n",
      "Job <400809> is submitted to default queue <lowpriority>.\n",
      "Job <400810> is submitted to default queue <lowpriority>.\n",
      "Job <400811> is submitted to default queue <lowpriority>.\n",
      "Job <400812> is submitted to default queue <lowpriority>.\n",
      "Job <400813> is submitted to default queue <lowpriority>.\n",
      "Job <400814> is submitted to default queue <lowpriority>.\n",
      "Job <400815> is submitted to default queue <lowpriority>.\n",
      "Job <400816> is submitted to default queue <lowpriority>.\n",
      "Job <400817> is submitted to default queue <lowpriority>.\n",
      "Job <400818> is submitted to default queue <lowpriority>.\n",
      "Job <400819> is submitted to default queue <lowpriority>.\n",
      "Job <400820> is submitted to default queue <lowpriority>.\n",
      "Job <400821> is submitted to default queue <lowpriority>.\n",
      "Job <400822> is submitted to default queue <lowpriority>.\n",
      "Job <400823> is submitted to default queue <lowpriority>.\n",
      "Job <400824> is submitted to default queue <lowpriority>.\n",
      "Job <400825> is submitted to default queue <lowpriority>.\n",
      "Job <400826> is submitted to default queue <lowpriority>.\n",
      "Job <400827> is submitted to default queue <lowpriority>.\n",
      "Job <400828> is submitted to default queue <lowpriority>.\n",
      "Job <400829> is submitted to default queue <lowpriority>.\n",
      "Job <400830> is submitted to default queue <lowpriority>.\n",
      "Job <400831> is submitted to default queue <lowpriority>.\n",
      "Job <400832> is submitted to default queue <lowpriority>.\n",
      "Job <400833> is submitted to default queue <lowpriority>.\n",
      "Job <400834> is submitted to default queue <lowpriority>.\n",
      "Job <400835> is submitted to default queue <lowpriority>.\n",
      "Job <400836> is submitted to default queue <lowpriority>.\n",
      "Job <400837> is submitted to default queue <lowpriority>.\n",
      "Job <400838> is submitted to default queue <lowpriority>.\n",
      "Job <400839> is submitted to default queue <lowpriority>.\n",
      "Job <400840> is submitted to default queue <lowpriority>.\n",
      "Job <400841> is submitted to default queue <lowpriority>.\n",
      "Job <400842> is submitted to default queue <lowpriority>.\n",
      "Job <400843> is submitted to default queue <lowpriority>.\n",
      "Job <400844> is submitted to default queue <lowpriority>.\n",
      "Job <400845> is submitted to default queue <lowpriority>.\n",
      "Job <400846> is submitted to default queue <lowpriority>.\n",
      "Job <400847> is submitted to default queue <lowpriority>.\n",
      "Job <400848> is submitted to default queue <lowpriority>.\n",
      "Job <400849> is submitted to default queue <lowpriority>.\n",
      "Job <400850> is submitted to default queue <lowpriority>.\n",
      "Job <400851> is submitted to default queue <lowpriority>.\n",
      "Job <400852> is submitted to default queue <lowpriority>.\n",
      "Job <400853> is submitted to default queue <lowpriority>.\n",
      "Job <400854> is submitted to default queue <lowpriority>.\n",
      "Job <400855> is submitted to default queue <lowpriority>.\n",
      "Job <400856> is submitted to default queue <lowpriority>.\n",
      "Job <400857> is submitted to default queue <lowpriority>.\n",
      "Job <400858> is submitted to default queue <lowpriority>.\n",
      "Job <400859> is submitted to default queue <lowpriority>.\n",
      "Job <400860> is submitted to default queue <lowpriority>.\n",
      "Job <400861> is submitted to default queue <lowpriority>.\n",
      "Job <400862> is submitted to default queue <lowpriority>.\n",
      "Job <400863> is submitted to default queue <lowpriority>.\n",
      "Job <400864> is submitted to default queue <lowpriority>.\n",
      "Job <400865> is submitted to default queue <lowpriority>.\n",
      "Job <400866> is submitted to default queue <lowpriority>.\n",
      "Job <400867> is submitted to default queue <lowpriority>.\n",
      "Job <400868> is submitted to default queue <lowpriority>.\n",
      "Job <400869> is submitted to default queue <lowpriority>.\n",
      "Job <400870> is submitted to default queue <lowpriority>.\n",
      "Job <400871> is submitted to default queue <lowpriority>.\n",
      "Job <400872> is submitted to default queue <lowpriority>.\n",
      "Job <400873> is submitted to default queue <lowpriority>.\n",
      "Job <400874> is submitted to default queue <lowpriority>.\n",
      "Job <400875> is submitted to default queue <lowpriority>.\n",
      "Job <400876> is submitted to default queue <lowpriority>.\n",
      "Job <400877> is submitted to default queue <lowpriority>.\n",
      "Job <400878> is submitted to default queue <lowpriority>.\n",
      "Job <400879> is submitted to default queue <lowpriority>.\n",
      "Job <400880> is submitted to default queue <lowpriority>.\n",
      "Job <400881> is submitted to default queue <lowpriority>.\n",
      "Job <400882> is submitted to default queue <lowpriority>.\n",
      "Job <400883> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "bams=$(ls /powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/*.bam)\n",
    "INTERVALS=\"/workspace/hramzr/2_Phd_PROJECT/WGS/RTC\"\n",
    "OUT=\"/workspace/hramzr/2_Phd_PROJECT/WGS/IR\"\n",
    "LOG=\"${OUT}/logs\"\n",
    "GATK=\"/powerplant/workspace/cflcyd/software/GATK/gatk-3.8.1/GenomeAnalysisTK.jar\"\n",
    "JAVA=\"java -jar -Xmx64G\"\n",
    "ref=\"/workspace/hramzr/2_Phd_PROJECT/snapper_genome/Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta\"\n",
    "for bam in ${bams}\n",
    "    do\n",
    "        NAME=`basename ${bam}`\n",
    "        PREFIX2=`echo ${NAME} | awk -F[_] '{print $1}'`\n",
    "        COMMAND=\"${JAVA} ${GATK} -T IndelRealigner \\\n",
    "                     -R ${ref} \\\n",
    "                     -I ${bam} \\\n",
    "                     -maxIsize 780 \\\n",
    "                     -targetIntervals ${INTERVALS}/${PREFIX2}.intervals \\\n",
    "                     -o ${OUT}/${PREFIX2}_IR.bam\"\n",
    "        # echo ${COMMAND}\n",
    "        bsub -o ${LOG}/${PREFIX2}.out -e ${LOG}/${PREFIX2}.err  \\\n",
    "              -J IR \\\n",
    "              -n 3 \\\n",
    "              -R \"rusage[mem=48000] span[hosts=1]\" \\\n",
    "              ${COMMAND} \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/008821357_M.bam /powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/967038331_M.bam\n",
      "java -jar -Xmx64G /powerplant/workspace/cflcyd/software/GATK/gatk-3.8.1/GenomeAnalysisTK.jar -T IndelRealigner -R /workspace/hramzr/2_Phd_PROJECT/snapper_genome/Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta -I /powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/008821357_M.bam -maxIsize 780 -targetIntervals /workspace/hramzr/2_Phd_PROJECT/WGS/RTC/008821357.intervals -o /workspace/hramzr/2_Phd_PROJECT/WGS/IR/008821357_IR.bam\n",
      "Job <401518> is submitted to default queue <lowpriority>.\n",
      "java -jar -Xmx64G /powerplant/workspace/cflcyd/software/GATK/gatk-3.8.1/GenomeAnalysisTK.jar -T IndelRealigner -R /workspace/hramzr/2_Phd_PROJECT/snapper_genome/Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta -I /powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/967038331_M.bam -maxIsize 780 -targetIntervals /workspace/hramzr/2_Phd_PROJECT/WGS/RTC/967038331.intervals -o /workspace/hramzr/2_Phd_PROJECT/WGS/IR/967038331_IR.bam\n",
      "Job <401519> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "bams=$(ls /powerplant/workspace/hramzr/2_Phd_PROJECT/WGS/mark_duplicates/{*008821357*.bam,*967038331*.bam})\n",
    "INTERVALS=\"/workspace/hramzr/2_Phd_PROJECT/WGS/RTC\"\n",
    "OUT=\"/workspace/hramzr/2_Phd_PROJECT/WGS/IR\"\n",
    "LOG=\"${OUT}/logs\"\n",
    "GATK=\"/powerplant/workspace/cflcyd/software/GATK/gatk-3.8.1/GenomeAnalysisTK.jar\"\n",
    "JAVA=\"java -jar -Xmx64G\"\n",
    "ref=\"/workspace/hramzr/2_Phd_PROJECT/snapper_genome/Chrysophrys_auratus.v.1.0.chromosomes.male.map.fasta\"\n",
    "echo $bams\n",
    "for bam in ${bams}\n",
    "    do\n",
    "        NAME=`basename ${bam}`\n",
    "        PREFIX2=`echo ${NAME} | awk -F[_] '{print $1}'`\n",
    "        COMMAND=\"${JAVA} ${GATK} -T IndelRealigner \\\n",
    "                     -R ${ref} \\\n",
    "                     -I ${bam} \\\n",
    "                     -maxIsize 780 \\\n",
    "                     -targetIntervals ${INTERVALS}/${PREFIX2}.intervals \\\n",
    "                     -o ${OUT}/${PREFIX2}_IR.bam\"\n",
    "        echo ${COMMAND}\n",
    "        bsub -o ${LOG}/malf.out -e ${LOG}/malf.err  \\\n",
    "              -J IR \\\n",
    "              -n 3 \\\n",
    "              -R \"rusage[mem=48000] span[hosts=1]\" \\\n",
    "              ${COMMAND} \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "---------- /software/OSutils/modules-4.4.1/share/Modules/modulefiles -----------\n",
      "1SynChro/1.15\n",
      "3d-dna/180922\n",
      "A5-miseq/20160825\n",
      "abyss/2.0.2\n",
      "abyss/2.1.0\n",
      "abyss/2.1.1\n",
      "abyss/2.1.2\n",
      "abyss/2.1.3\n",
      "abyss/2.1.4\n",
      "abyss/2.1.5\n",
      "adapterremoval/2.2.2\n",
      "AfterQC/0.9.1\n",
      "AfterQC/0.9.5\n",
      "allpathslg/44837\n",
      "allpathslg/44913\n",
      "allpathslg/49722\n",
      "allpathslg/50191\n",
      "allpathslg/52488\n",
      "amos/3.1.0\n",
      "angsd/0.913\n",
      "angsd/0.917\n",
      "angsd/0.917-116-g5d087b2\n",
      "angsd/0.918\n",
      "angsd/0.919\n",
      "angsd/0.921\n",
      "angsd/0.922\n",
      "angsd/0.923\n",
      "angsd/0.925\n",
      "angsd-wrapper/7224db2\n",
      "anytag/2.5.2\n",
      "apsim/7.10-r49ace54f9c8a670190aef9d8d0fb9d5477bb1534\n",
      "apsim/7.9-r4047\n",
      "apsimx/2018.01.30.2253\n",
      "apsimx/2018.09.28.3099\n",
      "apsimx/2019.01.08.3392\n",
      "apsimx/2019.04.03.3693\n",
      "apsimx/2019.06.05.3920\n",
      "apsimx/2019.07.18.4025\n",
      "apsimx/2019.10.04.4236\n",
      "apsimx/2020.04.09.5012\n",
      "apsimx/2020.08.04.5350\n",
      "arcs/1.0.6\n",
      "art/MountRainier\n",
      "aspera-cli/3.7.2.354.010c3b8\n",
      "assemblathon_stats/14dfdab\n",
      "assemblathon_stats/2011_10_13\n",
      "asub/2.1\n",
      "augustus/3.2.2\n",
      "augustus/3.3\n",
      "augustus/3.3.1\n",
      "augustus/3.3.3-new\n",
      "aws-cli/2.0.43\n",
      "bam2fastx/1.3.0\n",
      "bambam/1.4\n",
      "bamtools/2.3.0\n",
      "bamtools/2.4.0\n",
      "bats/0.4.0\n",
      "BBMap/37.28\n",
      "BBMap/37.53\n",
      "BBMap/37.93\n",
      "BBMap/38.33\n",
      "bcftools/1.10.2\n",
      "bcftools/1.3.1\n",
      "bcftools/1.6\n",
      "bcftools/1.9\n",
      "bcgTree/1.0.8\n",
      "beagle/4.0\n",
      "beagle/4.1\n",
      "BEAST/1.8.4\n",
      "BEAST2/2.4.7\n",
      "bedops/2.4.39\n",
      "bedtools/2.20.1\n",
      "bedtools/2.21.0\n",
      "bedtools/2.27.1\n",
      "bfc-submitted/v1\n",
      "biobambam/0.0.191\n",
      "biobambam2/2.0.106\n",
      "biobambam2/2.0.44\n",
      "bismark/0.19.1\n",
      "bismark/0.20.0\n",
      "blasr/5.2.badc092\n",
      "blast/2.2.26\n",
      "BLAT/36\n",
      "blobtools/1.0\n",
      "boost/1_61_0\n",
      "boost/1_61_0-dev\n",
      "boost/1_65_1\n",
      "boost/1_65_1-dev\n",
      "bowtie/1.0.0\n",
      "bowtie2/2.2.5\n",
      "bowtie2/2.2.9\n",
      "bowtie2/2.3.4.3\n",
      "BRAKER/1.11\n",
      "BRAKER/1.9\n",
      "BRAKER/2.0.1\n",
      "BRAKER/2.0.4\n",
      "BRAKER/2.0.5\n",
      "BRAKER/2.1.0\n",
      "BRAKER/2.1.2\n",
      "BRAKER/2.1.5\n",
      "breakdancer/4e44b43\n",
      "BSseeker2/2.1.5\n",
      "BUSCO/v1.2\n",
      "BUSCO/v2.0\n",
      "BUSCO/v3.0.2\n",
      "BUSCO/v4.1.2\n",
      "bwa/0.7.12\n",
      "bwa/0.7.15\n",
      "bwa/0.7.17\n",
      "canu/1.3\n",
      "canu/1.5\n",
      "canu/1.6\n",
      "canu/1.7\n",
      "cd-hit/4.6.1\n",
      "cdo/1.7.0\n",
      "cdo/1.9.3\n",
      "cdo/1.9.5\n",
      "cegma/v2.4.010312\n",
      "cegma/v2.5\n",
      "checkm/1.0.10\n",
      "checkm/1.0.11\n",
      "checkm/1.0.12\n",
      "checkm/1.0.13\n",
      "checkm/1.0.6\n",
      "checkm/1.0.7\n",
      "checkm/1.0.8\n",
      "checkm/1.1.3\n",
      "chromonomer/1.07\n",
      "chromosomer/0.1.3\n",
      "cndsrc/2010.10.11\n",
      "cndsrc/2017.11.30\n",
      "cnvnator/0.3.3\n",
      "cnvnator/0.4.1\n",
      "CodingQuarry/2.0\n",
      "COGNATE/v1.0\n",
      "colordiff/1.0.18\n",
      "conda/4.5.1\n",
      "conda/4.8.2\n",
      "cookiecutter/1.0.0\n",
      "crema/173e9a4\n",
      "crisp/c6cd86e\n",
      "crossmap/0.3.3\n",
      "csvkit/1.0.2\n",
      "cufflinks/2.2.1\n",
      "cutadapt/1.10\n",
      "cutadapt/1.15\n",
      "dammit/1.2\n",
      "data-registry/1.0\n",
      "data-registry/1.0-dev\n",
      "data-registry/1.0-test\n",
      "deepvariant/0.7.2-py27h5d9141f_1\n",
      "diamond/0.9.24\n",
      "Disco/1.1-beta\n",
      "docker-compose/1.11.2\n",
      "docker-compose/1.7.1\n",
      "dtool/3.25.0\n",
      "dvc/1.6.1\n",
      "DWGSIM/0561546\n",
      "ea-utils/1.04.807\n",
      "ea-utils/1.1.2-484\n",
      "ea-utils/1.1.2-537\n",
      "ea-utils/1.1.2-806\n",
      "eddypro-engine/5.1.1\n",
      "eddypro-engine/5.2.0\n",
      "eddypro-engine/5.2.1\n",
      "eddypro-engine/6.0.0\n",
      "eddypro-engine/6.1.0\n",
      "eddypro-engine/6.2.0\n",
      "eddypro-engine/6.2.1\n",
      "edta/1.8.3\n",
      "EffectorP/1.0\n",
      "eg-all/91\n",
      "eg-web-bacteria/91\n",
      "eg-web-common/91\n",
      "eg-web-fungi/91\n",
      "eg-web-metazoa/91\n",
      "eg-web-plants/91\n",
      "eg-web-protists/91\n",
      "eg-web-search/91\n",
      "eg-web-testsuite/91\n",
      "ehive/2.4\n",
      "eigensoft/7.2.1\n",
      "eman2/2.3\n",
      "emboss/6.4.0\n",
      "emboss/6.5.7\n",
      "emboss/6.6.0\n",
      "emirge/0.61.1\n",
      "ensembl/69\n",
      "ensembl/91\n",
      "ensembl/92\n",
      "ensembl/93\n",
      "ensembl-analysis/69\n",
      "ensembl-analysis/91\n",
      "ensembl-analysis/92\n",
      "ensembl-analysis/93\n",
      "ensembl-api/69\n",
      "ensembl-api/91\n",
      "ensembl-api/92\n",
      "ensembl-api/93\n",
      "ensembl-compara/69\n",
      "ensembl-compara/91\n",
      "ensembl-compara/92\n",
      "ensembl-compara/93\n",
      "ensembl-funcgen/69\n",
      "ensembl-funcgen/91\n",
      "ensembl-funcgen/92\n",
      "ensembl-funcgen/93\n",
      "ensembl-genebuild/69\n",
      "ensembl-genebuild/91\n",
      "ensembl-genebuild/92\n",
      "ensembl-genebuild/93\n",
      "ensemblgenomes-api/91\n",
      "ensembl-git-tools/d55068a\n",
      "ensembl-hive/69\n",
      "ensembl-hive/91\n",
      "ensembl-hive/92\n",
      "ensembl-hive/93\n",
      "ensembl-hive-openlava/69\n",
      "ensembl-hive-openlava/91\n",
      "ensembl-hive-openlava/92\n",
      "ensembl-hive-openlava/93\n",
      "ensembl-io/69\n",
      "ensembl-io/91\n",
      "ensembl-io/92\n",
      "ensembl-io/93\n",
      "ensembl-killlist/69\n",
      "ensembl-killlist/91\n",
      "ensembl-killlist/92\n",
      "ensembl-killlist/93\n",
      "ensembl-metadata/69\n",
      "ensembl-metadata/91\n",
      "ensembl-metadata/92\n",
      "ensembl-metadata/93\n",
      "ensembl-notebook/latest\n",
      "ensembl-orm/69\n",
      "ensembl-orm/91\n",
      "ensembl-orm/92\n",
      "ensembl-orm/93\n",
      "ensembl-pipeline/69\n",
      "ensembl-pipeline/91\n",
      "ensembl-pipeline/92\n",
      "ensembl-pipeline/93\n",
      "ensembl-production/69\n",
      "ensembl-production/91\n",
      "ensembl-production/92\n",
      "ensembl-production/93\n",
      "ensembl-taxonomy/69\n",
      "ensembl-taxonomy/91\n",
      "ensembl-taxonomy/92\n",
      "ensembl-taxonomy/93\n",
      "ensembl-tools/69\n",
      "ensembl-tools/91\n",
      "ensembl-tools/92\n",
      "ensembl-tools/93\n",
      "ensembl-variation/69\n",
      "ensembl-variation/91\n",
      "ensembl-variation/92\n",
      "ensembl-variation/93\n",
      "ensembl-vep/69\n",
      "ensembl-vep/91\n",
      "ensembl-vep/92\n",
      "ensembl-vep/93\n",
      "entrez-direct/13.8.20200819\n",
      "e-PCR/2.3.9\n",
      "ericscript/0.5.5\n",
      "e-utilities/5.40\n",
      "EVidenceModeler/1.1.1\n",
      "evigene/02-11-2017\n",
      "evigene/18-01-2018\n",
      "evigene/18-09-2013\n",
      "exonerate/2.4.0\n",
      "FALCON/1.7.4\n",
      "fasta/35.4.12\n",
      "fasta/36.3.8g\n",
      "fast-gbs/2016.12.14\n",
      "fast-gbs/2017.01.12\n",
      "fastp/0.12.5\n",
      "fastp/0.19.4\n",
      "fastp/0.20.0\n",
      "FastQC/0.11.1\n",
      "FastQC/0.11.2\n",
      "FastQC/0.11.7\n",
      "fastq_screen/v0.11.1\n",
      "fastq_screen/v0.14.0\n",
      "fastq_screen/v0.4.4\n",
      "fastq_screen/v0.5.2\n",
      "faststructure/1.0\n",
      "FastTree/2.1.10\n",
      "fastx_toolkit/0.0.13\n",
      "FGMP/1.0\n",
      "FLASH/1.2.11\n",
      "flye/2.4.2\n",
      "flye/2.5\n",
      "flye/2.7.1\n",
      "freebayes/1.1.0\n",
      "freebayes/1.3.1\n",
      "freebayes/v1.0.2-16-gd466dde\n",
      "freetype/2.9\n",
      "GapFiller/v1.10\n",
      "GARM/0.7.4\n",
      "gatk/1.0\n",
      "gatk/3.7\n",
      "gatk/3.8.0\n",
      "Gblocks/0.91b\n",
      "gdal/2.4.0\n",
      "GEM/20130406-045632\n",
      "gemoma/1.4.2\n",
      "gemoma/1.6.4\n",
      "geneid/v1.4.4\n",
      "GeneMark/4.21\n",
      "GeneMark/4.33\n",
      "GeneMark/4.56\n",
      "genomethreader/1.7.0\n",
      "genometools/1.5.10\n",
      "getorganelle/1.6.2e\n",
      "gffread/0.9.6\n",
      "git/2.11.0\n",
      "git/2.12.2\n",
      "git/2.14.1\n",
      "git/2.21.0\n",
      "git-annex/6.20160613-g35dbe35\n",
      "gitflow/15aab26\n",
      "git-lfs/2.3.0\n",
      "globus-cli/1.2.2\n",
      "globus-cli/1.6.0\n",
      "gmap/2016-06-09\n",
      "gmap/2017-06-20\n",
      "gmap/2017-09-30\n",
      "gmap/2018-03-01\n",
      "gmap/2018-03-25\n",
      "gmap/2018-05-11\n",
      "gmap/2018-05-30\n",
      "gmato/v1.2\n",
      "GMcloser/1.6.2\n",
      "golang/1.14.2\n",
      "golang/1.9.7\n",
      "GroIMP/1.5\n",
      "gromacs/2018\n",
      "gromacs/2018.1\n",
      "gromacs/2018.2\n",
      "gromacs/2018.3\n",
      "gromacs/2018.4\n",
      "gromacs/2018.5\n",
      "gromacs/2019\n",
      "guppy/3.2.4\n",
      "guppy/3.4.4\n",
      "guppy/3.5.2\n",
      "guppy/3.6.1\n",
      "hagfish/05d08d1\n",
      "hagfish/1_May_2013\n",
      "hapcol/97d4a5e\n",
      "hapcompass/0.8.2\n",
      "HapCUT2/0121162\n",
      "Haplosim/ae652a2\n",
      "Haplosim/bd39c6f\n",
      "HapTree/1.0\n",
      "hdf5/1.8.17\n",
      "HiC-pro/2.11.1\n",
      "hicup/0.5.9\n",
      "hicup/0.7.2\n",
      "hisat2/2.1.0\n",
      "hmmer/3.0\n",
      "hmmer/3.1b1\n",
      "hmmer/3.1b2\n",
      "hmmer/3.3\n",
      "HTSeq/0.6.1\n",
      "HTSeq/0.7.2\n",
      "HTSeq/0.9.1\n",
      "htslib/1.10.2\n",
      "htslib/1.3.1\n",
      "htslib/1.7\n",
      "htslib/1.9\n",
      "hub/2.8.4\n",
      "humann2/0.11.1\n",
      "iCommands/4.1.8\n",
      "idba/a1bafe6\n",
      "IGV/2.3.93\n",
      "infernal/1.1.2\n",
      "iqtree/1.6.11\n",
      "JAGS/4.0.0\n",
      "JAGS/4.2.0\n",
      "java/1.7.0\n",
      "java/1.8.0\n",
      "jcvi/0.7.3\n",
      "jcvi/0.7.7\n",
      "jcvi/0.8.12\n",
      "jellyfish/1.1.10\n",
      "jellyfish/2.1.3\n",
      "jellyfish/2.2.10\n",
      "jmodeltest/2.1.10\n",
      "jq/1.6\n",
      "kat/2.3.1\n",
      "kat/2.4.1\n",
      "kentUtils/302.1\n",
      "khmer/2.0\n",
      "KMA/1.2.4\n",
      "KMC/3.0.0\n",
      "kneaddata/0.6.1\n",
      "kraken/0.10.5\n",
      "kraken/1.0\n",
      "kraken/1.1\n",
      "Krona/2.7\n",
      "LACHESIS/20150929\n",
      "last/770\n",
      "lastz/1.02.00\n",
      "LDhat/81596e2\n",
      "links/v1.8.5\n",
      "links/v1.8.6\n",
      "LocHap/2.0\n",
      "luarocks/2.4.1\n",
      "lumpy-sv/0.2.13\n",
      "MACS2/2.1.1\n",
      "mafft/7.307\n",
      "mandalorion-episode-ii/6219d58\n",
      "mapcomp/1.0\n",
      "mapgd/d3edee2\n",
      "MAPGD/0.4.26\n",
      "MAPGD/0.5\n",
      "mason/0.1.2\n",
      "MaSuRCA/3.2.2\n",
      "MaSuRCA/3.2.3\n",
      "MaSuRCA/3.2.4\n",
      "MaSuRCA/3.2.8\n",
      "MaSuRCA/3.3.3\n",
      "mauve/2015.02.13\n",
      "mcl/14-137\n",
      "mcscan/0.8\n",
      "MECAT/3898797\n",
      "megahit/1.1.1\n",
      "megahit/1.1.3\n",
      "megahit/1.2.6\n",
      "megahit/1.2.9\n",
      "meme/4.11.2\n",
      "Meraculous/2.2.4\n",
      "Metaassembler/1.5\n",
      "metabat/0.32.4\n",
      "metabat2/2.15\n",
      "metabat2/2.15-3-g367a7ef\n",
      "metaphlan2/2.6.0\n",
      "microbiome_helper/4330e63\n",
      "mikado/1.1\n",
      "miniasm/0.2\n",
      "miniasm/55cf018\n",
      "miniconda2/4.3.13\n",
      "minimap2/2.17\n",
      "minimap2/2.9\n",
      "mira/4.0.2\n",
      "miRanda/3.3a\n",
      "mitoz/2.3\n",
      "mplus/0.1\n",
      "mrbayes/3.2.7\n",
      "mrbayes/3.2.7a\n",
      "mrbayes/3.2.7a-gpu\n",
      "msms/3.2rc_163\n",
      "multiqc/1.0\n",
      "multiqc/1.2\n",
      "multiqc/1.3\n",
      "multiqc/1.5\n",
      "multiqc/1.6\n",
      "multiqc/1.7\n",
      "multiqc/1.8\n",
      "multiqc/1.9\n",
      "MUMmer/3.23\n",
      "muscle/3.8.1551\n",
      "muscle/4.0.170\n",
      "nanopack/1.0.0\n",
      "ncbi-blast/2.10.1\n",
      "ncbi-blast/2.2.25\n",
      "ncbi-blast/2.2.30\n",
      "ncbi-blast/2.6.0\n",
      "ncbi-genome-download/0.2.12\n",
      "NetLogo/6.0.4\n",
      "newbler/2.5.3\n",
      "newbler/2.7\n",
      "newbler/2.9\n",
      "nextclip/v1.3.1\n",
      "nextgenmap/v0.5.5\n",
      "NextGenMap/v0.5.5\n",
      "ngmlr/0.2.7\n",
      "NovoGraph/1.0.0\n",
      "NOVOPlasty/2.7.2\n",
      "NPStat/v1\n",
      "nQuire/a990a88\n",
      "NSEG/2018-05-30\n",
      "NxTrim/0.4.2\n",
      "NxTrim/v0.4.1-0f17575\n",
      "oases/0.2.09\n",
      "obitools/1.2.11\n",
      "octave/4.2.2\n",
      "octave/4.4.1\n",
      "openbugs/3.2.3\n",
      "opendronemap/0.4.0\n",
      "opendronemap/0.4.1\n",
      "openlava/3.2\n",
      "openmpi/1.8.8\n",
      "openmpi/4.0.2\n",
      "OPERA-LG/v2.0.6\n",
      "orthofinder/2.3.1\n",
      "orthomcl/2.0.9\n",
      "PAML/4.9i\n",
      "pandaseq/2.10\n",
      "pandoc/1.19.2\n",
      "PASA/2.0.2\n",
      "PASA/2.1.0\n",
      "PASA/2.2.0\n",
      "PASA/2.3.3\n",
      "patman/1.2.2\n",
      "pblat/2.0\n",
      "pblat/2.1\n",
      "PBSIM/308622d\n",
      "PBSuite/15.8.24\n",
      "PBSuite/17.05.09-dbrowneup\n",
      "pcl/1.8.1\n",
      "pcl/1.9.0\n",
      "pcl/1.9.1\n",
      "PedigreeSim/v2.0\n",
      "perl/5.10.1\n",
      "perl/5.10.1-thread-multi\n",
      "perl/5.12.5\n",
      "perl/5.12.5-thread-multi\n",
      "perl/5.14.4\n",
      "perl/5.14.4-thread-multi\n",
      "perl/5.16.3\n",
      "perl/5.16.3-thread-multi\n",
      "perl/5.18.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "perl/5.20.3\n",
      "perl/5.20.3-thread-multi\n",
      "perl/5.22.2\n",
      "perl/5.22.2-thread-multi\n",
      "perl/5.24.0\n",
      "perl/5.24.0-thread-multi\n",
      "perl/5.24.1\n",
      "perl/5.24.1-thread-multi\n",
      "perl/5.26.0\n",
      "perl/5.26.0-thread-multi\n",
      "perl/5.28.0\n",
      "perl/5.28.0-thread-multi\n",
      "perl/5.30.2\n",
      "perl/5.30.2-thread-multi\n",
      "perl/5.8.9\n",
      "perl/5.8.9-thread-multi\n",
      "perl/bio-extras\n",
      "perl/bio-extras-5.24.0-legacy\n",
      "perl/core\n",
      "perl/ensembl\n",
      "perl/jupyterhub\n",
      "perlbrew/0.76\n",
      "pfr-python2/2.7.13\n",
      "pfr-python3/3.6.1\n",
      "pfr-python3/3.6.5\n",
      "pfr-python3/3.6.6\n",
      "phylip/3.696\n",
      "phylobayes/4.1c\n",
      "PhyloSift/1.0.1\n",
      "picard-tools/1.112\n",
      "picard-tools/1.79\n",
      "picard-tools/2.10.1\n",
      "picard-tools/2.18.7\n",
      "picard-tools/2.2.4\n",
      "picard-tools/2.9.4\n",
      "picrust2/2.3.0\n",
      "pilon/1.18\n",
      "pilon/1.20\n",
      "pilon/1.23\n",
      "pinfish/0.1.0\n",
      "platanus/1.2.4\n",
      "platanus_trim/1.0.7\n",
      "Platypus/0.8.1\n",
      "plink/1.07\n",
      "plink/1.90b6.5\n",
      "plink2/v2.00a2lm\n",
      "plncpro/1.1\n",
      "popoolation2/1201\n",
      "porechop/0.2.3\n",
      "portcullis/1.1.0\n",
      "portcullis/1.1.1\n",
      "portcullis/1.1.2\n",
      "powerPlant/core\n",
      "powerPlant/locate\n",
      "pplacer/v1.1.alpha17-6-g5cecf99\n",
      "ppsPCP/1.0\n",
      "primer3/2.2.3\n",
      "primer3/2.3.5\n",
      "prinseq-lite/0.20.4\n",
      "Prodigal/2.6.3\n",
      "proj/5.2.0\n",
      "prokka/1.11\n",
      "prokka/1.13\n",
      "proteinortho/6.0.15\n",
      "purge_haplotigs/1.1.0\n",
      "pyarrow/0.16.0\n",
      "pycoqc/2.5.0.21\n",
      "qiime/1.9.1\n",
      "qiime2/2018.11\n",
      "qiime2/2018.2\n",
      "qiime2/2019.1\n",
      "qiime2/2019.10\n",
      "qiime2/2019.1-picrust2\n",
      "qiime2/2019.4\n",
      "qiime2/2019.7\n",
      "qiime2/2019.7-picrust2\n",
      "qiime2/2020.2\n",
      "qiime2/2020.6\n",
      "qualimap/v2.2.1\n",
      "quast/4.2\n",
      "R/3.3.0\n",
      "R/3.3.1\n",
      "R/3.3.3\n",
      "R/3.4.0\n",
      "R/3.4.1\n",
      "R/3.4.2\n",
      "R/3.4.3\n",
      "R/3.4.4\n",
      "R/3.5.0\n",
      "R/3.5.2\n",
      "R/3.5.3\n",
      "R/3.6.1\n",
      "R/3.6.2\n",
      "R/3.6.3\n",
      "R/4.0.0\n",
      "racon/1.3.0\n",
      "racon/1.3.1\n",
      "racon/1.3.2\n",
      "racon/1.3.3\n",
      "racon/1.4.0\n",
      "racon/1.4.2\n",
      "racon/1.4.3\n",
      "racon/1.4.7\n",
      "ragoo/1.01\n",
      "ragoo/1.02\n",
      "RAILS/v1.2\n",
      "RANBOW/1\n",
      "RANBOW/1-temp\n",
      "RECON/1.08\n",
      "REPCLASS/1.0.1\n",
      "RepeatMasker-open/3-3-0\n",
      "RepeatMasker-open/4-0-3\n",
      "RepeatMasker-open/4-0-5\n",
      "RepeatModeler/1.0.8\n",
      "RepeatModeler-open/1.0.11\n",
      "RepeatScout/1.0.5\n",
      "REPET/2.5\n",
      "rnacocktail/0.2.1\n",
      "RSEM/1.3.0\n",
      "rtg-tools/3.7.1-eb13bbb\n",
      "ruby/2.1.9\n",
      "ruby/2.2.7\n",
      "ruby/2.3.4\n",
      "ruby/2.4.1\n",
      "sabre/1.000\n",
      "salmon/1.1.0\n",
      "salmon/1.3.0\n",
      "salsa/v2.2\n",
      "samtools/0.1.18\n",
      "samtools/0.1.19\n",
      "samtools/1.1\n",
      "samtools/1.10\n",
      "samtools/1.2\n",
      "samtools/1.3.1\n",
      "samtools/1.7\n",
      "samtools/1.9\n",
      "ScanIndel/1.2dev\n",
      "SDhap/56d7678\n",
      "seqkit/0.7.0\n",
      "seqmagick/0.8.0\n",
      "seqtk/1.2\n",
      "sex-detector-plusplus/00f7d723\n",
      "sga/0.10.10-2\n",
      "sga/0.10.13\n",
      "sga/0.10.15\n",
      "sga/0.9.19\n",
      "sga/0.9.35\n",
      "SGSGeneLoss/0.1\n",
      "singularity/2.4.2\n",
      "singularity/2.5.1\n",
      "singularity/3\n",
      "SLiM/3.1\n",
      "smalt/0.7.6\n",
      "snakemake/5.10.0\n",
      "snap/1.0beta.18\n",
      "snapr/0.16alpha.51\n",
      "sniffles/1.0.11\n",
      "SnpEff/4.3t\n",
      "SNPhylo/20140701\n",
      "SNPhylo/20160204\n",
      "SOAPdenovo/1.05\n",
      "SOAPdenovo/2.04\n",
      "SOAPdenovo-Trans/1.03\n",
      "SolexaQA/3.1.5\n",
      "sortmerna/2.1\n",
      "sortmerna/3.0.3\n",
      "sortmerna/4.2.0\n",
      "SPAdes/3.10.1\n",
      "SPAdes/3.11.1\n",
      "SPAdes/3.12.0\n",
      "SPAdes/3.13.0\n",
      "SPAdes/3.14.0\n",
      "SPAdes/3.7.1\n",
      "SqueezeMeta/0.4.4\n",
      "SqueezeMeta/1.0.0-beta\n",
      "sratoolkit/2.8.2.1\n",
      "SSPACE/2.0\n",
      "stacks/1.40\n",
      "stacks/1.40_withbam\n",
      "stacks/2.0\n",
      "stacks/2.1\n",
      "stacks/2.2\n",
      "stamp/2.1.3\n",
      "standard-RAxML/8.2.11\n",
      "standard-RAxML/8.2.9\n",
      "STAR/2.4.1d\n",
      "STAR/2.4.2a\n",
      "STAR/2.5.2a\n",
      "STAR/2.5.2b\n",
      "STAR/2.5.3a\n",
      "STAR/2.6.1a\n",
      "STAR/2.6.1d\n",
      "STAR-Fusion/0.8.0\n",
      "stringtie/1.3.3b.Linux_x86_64\n",
      "subread/1.4.5\n",
      "subread/1.5.3\n",
      "supernova/1.2.2\n",
      "survivor/1.0.5\n",
      "SV2/1.4.3.2\n",
      "svaba/1.1.3\n",
      "svim/1.2.0\n",
      "svtools/0.3.2\n",
      "swan/3516c2f\n",
      "tabix/0.2.6\n",
      "TargQC/1.4.4\n",
      "tasr/1.6.2\n",
      "tassel/3.0\n",
      "tassel/4.0\n",
      "tassel/5.0\n",
      "tassel/5.2.28\n",
      "taxmaps/0.2.1\n",
      "tetools/1.1\n",
      "texlive/20151117\n",
      "tigmint/1.1.2\n",
      "tomahawk/0.7.1\n",
      "tophat/2.0.13\n",
      "tophat/2.1.1\n",
      "totalview/2017.2.11\n",
      "TransDecoder/3.0.1\n",
      "TransDecoder/5.0.1\n",
      "TransDecoder/5.5.0\n",
      "transrate/1.0.3\n",
      "trf/4.07\n",
      "trim_galore/0.4.1\n",
      "trim_galore/0.4.3\n",
      "Trimmomatic/0.36\n",
      "trinityrnaseq/2.0.6\n",
      "trinityrnaseq/2.10.0\n",
      "trinityrnaseq/2.2.0\n",
      "trinityrnaseq/2.3.2\n",
      "trinityrnaseq/2.4.0\n",
      "trinityrnaseq/2.5.1\n",
      "trinityrnaseq/2.6.5\n",
      "trinityrnaseq/2.8.6\n",
      "trinityrnaseq/2.9.0\n",
      "trinityrnaseq/2.9.1\n",
      "trinotate/3.0.2\n",
      "trinotate/3.2.0\n",
      "tRNAscan-SE/1.3.1\n",
      "tRNAscan-SE/2.0\n",
      "ucsc-pslcdnafilter/324\n",
      "unicycler/0.4.8\n",
      "unixODBC/2.3.0\n",
      "vcflib/50a3229\n",
      "vcflib/586c5ae\n",
      "vcflib/73b7f13\n",
      "vcftools/0.1.12\n",
      "vcftools/0.1.14\n",
      "velvet/1.2.10\n",
      "VersioningService/69\n",
      "VersioningService/91\n",
      "VersioningService/92\n",
      "VersioningService/93\n",
      "vg/1.10.0\n",
      "vg/1.11.0\n",
      "vg/1.12.0\n",
      "vg/1.12.1\n",
      "vg/1.13.0\n",
      "vg/1.7.0\n",
      "vg/1.8.0\n",
      "vg/1.9.0\n",
      "ViennaRNA/2.2.2\n",
      "vsearch/2.3.2\n",
      "wgs/8.3rc2\n",
      "whatshap/0.14.1-py36_0\n",
      "wise/2.4.1\n",
      "wkhtmlox/0.12.3\n",
      "wtdbg2/2.2\n"
     ]
    }
   ],
   "source": [
    "module avail"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call structural variants DELLY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ModuleCmd_Load.c(213):ERROR:105: Unable to locate a modulefile for 'singularity/2.6.1'\n",
      "\u001b[31mFATAL:  \u001b[0m Image file already exists: \"dellySV.simg\" - will not overwrite\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# download docker for singularity\n",
    "module load singularity/2.6.1\n",
    "# singularity pull docker://sameerdcosta/parliament2\n",
    "# bsub \\\n",
    "# -J \"pull singularity docker\" \\\n",
    "cd /workspace/hramzr/2_Phd_PROJECT/VarCallingWGS/\n",
    "singularity pull --name dellySV.simg docker://dellytools/delly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "**********************************************************************\n",
      "Program: Delly\n",
      "This is free software, and you are welcome to redistribute it under\n",
      "certain conditions (BSD License); for license details use '-l'.\n",
      "This program comes with ABSOLUTELY NO WARRANTY; for details use '-w'.\n",
      "\n",
      "Delly (Version: 0.8.3)\n",
      "Contact: Tobias Rausch (rausch@embl.de)\n",
      "**********************************************************************\n",
      "\n",
      "Usage: delly <command> <arguments>\n",
      "\n",
      "Short-read commands:\n",
      "    call         discover and genotype structural variants\n",
      "    merge        merge structural variants across VCF/BCF files and within a single VCF/BCF file\n",
      "    filter       filter somatic or germline structural variants\n",
      "\n",
      "Long-read commands:\n",
      "    lr           long-read SV discovery\n",
      "\n",
      "Read-depth commands:\n",
      "    rd           read-depth normalization\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "singularity run dellySV.simg \\\n",
    "delly "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/workspace/hramzr/2_Phd_PROJECT/cnv_calling/delly/.: /workspace/hramzr/2_Phd_PROJECT/cnv_calling/delly/.: is a directory\n"
     ]
    },
    {
     "ename": "",
     "evalue": "126",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bash /workspace/hramzr/2_Phd_PROJECT/cnv_calling/delly/."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
