{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "---------- /software/OSutils/modules-4.4.1/share/Modules/modulefiles -----------\n",
      "1SynChro/1.15                    jellyfish/2.2.10\n",
      "3d-dna/180922                    jmodeltest/2.1.10\n",
      "A5-miseq/20160825                jq/1.6\n",
      "abyss/2.0.2                      kat/2.3.1\n",
      "abyss/2.1.0                      kat/2.4.1\n",
      "abyss/2.1.1                      kentUtils/302.1\n",
      "abyss/2.1.2                      khmer/2.0\n",
      "abyss/2.1.3                      KMA/1.2.4\n",
      "abyss/2.1.4                      KMC/3.0.0\n",
      "abyss/2.1.5                      kneaddata/0.6.1\n",
      "adapterremoval/2.2.2             kraken/0.10.5\n",
      "AfterQC/0.9.1                    kraken/1.0\n",
      "AfterQC/0.9.5                    kraken/1.1\n",
      "allpathslg/44837                 Krona/2.7\n",
      "allpathslg/44913                 LACHESIS/20150929\n",
      "allpathslg/49722                 last/770\n",
      "allpathslg/50191                 lastz/1.02.00\n",
      "allpathslg/52488                 LDhat/81596e2\n",
      "amos/3.1.0                       links/v1.8.5\n",
      "angsd/0.913                      links/v1.8.6\n",
      "angsd/0.917                      LocHap/2.0\n",
      "angsd/0.917-116-g5d087b2         luarocks/2.4.1\n",
      "angsd/0.918                      lumpy-sv/0.2.13\n",
      "angsd/0.919                      MACS2/2.1.1\n",
      "angsd/0.921                      mafft/7.307\n",
      "angsd/0.922                      mapcomp/1.0\n",
      "angsd/0.923                      mapgd/d3edee2\n",
      "angsd/0.925                      MAPGD/0.4.26\n",
      "angsd-wrapper/7224db2            MAPGD/0.5\n",
      "anytag/2.5.2                     mason/0.1.2\n",
      "apsim/7.9-r4047                  MaSuRCA/3.2.2\n",
      "apsim/7.9-r4047-docker           MaSuRCA/3.2.3\n",
      "apsim/7.9-r4176                  MaSuRCA/3.2.4\n",
      "apsimx/2018.01.30.2253           MaSuRCA/3.2.8\n",
      "apsimx/2018.09.28.3099           MaSuRCA/3.3.3\n",
      "apsimx/2019.01.08.3392           mauve/2015.02.13\n",
      "apsimx/2019.04.03.3693           mcl/14-137\n",
      "apsimx/2019.06.05.3920           mcscan/0.8\n",
      "apsimx/2019.07.18.4025           MECAT/3898797\n",
      "apsimx/2019.10.04.4236           megahit/1.1.1\n",
      "apsimx/2189-docker               megahit/1.1.3\n",
      "arcs/1.0.6                       megahit/1.2.6\n",
      "art/MountRainier                 meme/4.11.2\n",
      "aspera-cli/3.7.2.354.010c3b8     Meraculous/2.2.4\n",
      "assemblathon_stats/14dfdab       Metaassembler/1.5\n",
      "assemblathon_stats/2011_10_13    metabat/0.32.4\n",
      "asub/2.1                         metabat2/2.15\n",
      "augustus/3.2.2                   metabat2/2.15-3-g367a7ef\n",
      "augustus/3.3                     metaphlan2/2.6.0\n",
      "augustus/3.3.1                   microbiome_helper/4330e63\n",
      "augustus/3.3.3-new               mikado/1.1\n",
      "bam2fastx/1.3.0                  miniasm/0.2\n",
      "bambam/1.4                       miniasm/55cf018\n",
      "bamtools/2.3.0                   miniconda2/4.3.13\n",
      "bamtools/2.4.0                   minimap2/2.17\n",
      "bats/0.4.0                       minimap2/2.9\n",
      "BBMap/37.28                      mira/4.0.2\n",
      "BBMap/37.53                      miRanda/3.3a\n",
      "BBMap/37.93                      mitoz/2.3\n",
      "BBMap/38.33                      mplus/0.1\n",
      "bcftools/1.3.1                   mrbayes/3.2.7\n",
      "bcftools/1.6                     mrbayes/3.2.7a\n",
      "bcftools/1.9                     mrbayes/3.2.7a-gpu\n",
      "bcgTree/1.0.8                    msms/3.2rc_163\n",
      "beagle/4.0                       multiqc/1.0\n",
      "beagle/4.1                       multiqc/1.2\n",
      "BEAST/1.8.4                      multiqc/1.3\n",
      "BEAST2/2.4.7                     multiqc/1.5\n",
      "bedtools/2.20.1                  multiqc/1.6\n",
      "bedtools/2.21.0                  multiqc/1.7\n",
      "bedtools/2.27.1                  MUMmer/3.23\n",
      "bfc-submitted/v1                 muscle/3.8.1551\n",
      "biobambam/0.0.191                muscle/4.0.170\n",
      "biobambam2/2.0.106               nanopack/1.0.0\n",
      "biobambam2/2.0.44                ncbi-blast/2.2.25\n",
      "bismark/0.19.1                   ncbi-blast/2.2.30\n",
      "bismark/0.20.0                   ncbi-blast/2.6.0\n",
      "blasr/5.2.badc092                NetLogo/6.0.4\n",
      "blast/2.2.26                     newbler/2.5.3\n",
      "BLAT/36                          newbler/2.7\n",
      "blobtools/1.0                    newbler/2.9\n",
      "boost/1_61_0                     nextclip/v1.3.1\n",
      "boost/1_61_0-dev                 nextgenmap/v0.5.5\n",
      "boost/1_65_1                     NextGenMap/v0.5.5\n",
      "boost/1_65_1-dev                 ngmlr/0.2.7\n",
      "bowtie/1.0.0                     NovoGraph/1.0.0\n",
      "bowtie2/2.2.5                    NOVOPlasty/2.7.2\n",
      "bowtie2/2.2.9                    NPStat/v1\n",
      "bowtie2/2.3.4.3                  nQuire/a990a88\n",
      "BRAKER/1.11                      NSEG/2018-05-30\n",
      "BRAKER/1.9                       NxTrim/0.4.2\n",
      "BRAKER/2.0.1                     NxTrim/v0.4.1-0f17575\n",
      "BRAKER/2.0.4                     oases/0.2.09\n",
      "BRAKER/2.0.5                     obitools/1.2.11\n",
      "BRAKER/2.1.0                     octave/4.2.2\n",
      "BRAKER/2.1.2                     octave/4.4.1\n",
      "breakdancer/4e44b43              opendronemap/0.4.0\n",
      "BSseeker2/2.1.5                  opendronemap/0.4.1\n",
      "BUSCO/v1.2                       openlava/3.2\n",
      "BUSCO/v2.0                       openmpi/1.8.8\n",
      "BUSCO/v3.0.2                     openmpi/4.0.2\n",
      "bwa/0.7.12                       OPERA-LG/v2.0.6\n",
      "bwa/0.7.15                       orthofinder/2.3.1\n",
      "bwa/0.7.17                       orthomcl/2.0.9\n",
      "canu/1.3                         PAML/4.9i\n",
      "canu/1.5                         pandaseq/2.10\n",
      "canu/1.6                         pandoc/1.19.2\n",
      "canu/1.7                         PASA/2.0.2\n",
      "cd-hit/4.6.1                     PASA/2.1.0\n",
      "cdo/1.7.0                        PASA/2.2.0\n",
      "cdo/1.9.3                        PASA/2.3.3\n",
      "cdo/1.9.5                        patman/1.2.2\n",
      "cegma/v2.4.010312                pblat/2.0\n",
      "cegma/v2.5                       pblat/2.1\n",
      "checkm/1.0.10                    PBSIM/308622d\n",
      "checkm/1.0.11                    PBSuite/15.8.24\n",
      "checkm/1.0.12                    PBSuite/17.05.09-dbrowneup\n",
      "checkm/1.0.13                    pcl/1.8.1\n",
      "checkm/1.0.6                     pcl/1.9.0\n",
      "checkm/1.0.7                     pcl/1.9.1\n",
      "checkm/1.0.8                     PedigreeSim/v2.0\n",
      "chromonomer/1.07                 perl/5.10.1\n",
      "chromosomer/0.1.3                perl/5.10.1-thread-multi\n",
      "cndsrc/2010.10.11                perl/5.12.5\n",
      "cndsrc/2017.11.30                perl/5.12.5-thread-multi\n",
      "cnvnator/0.3.3                   perl/5.14.4\n",
      "cnvnator/0.4.1                   perl/5.14.4-thread-multi\n",
      "CodingQuarry/2.0                 perl/5.16.3\n",
      "COGNATE/v1.0                     perl/5.16.3-thread-multi\n",
      "colordiff/1.0.18                 perl/5.18.4\n",
      "conda/4.5.1                      perl/5.20.3\n",
      "cookiecutter/1.0.0               perl/5.20.3-thread-multi\n",
      "crema/173e9a4                    perl/5.22.2\n",
      "crisp/c6cd86e                    perl/5.22.2-thread-multi\n",
      "crossmap/0.3.3                   perl/5.24.0\n",
      "csvkit/1.0.2                     perl/5.24.0-thread-multi\n",
      "cufflinks/2.2.1                  perl/5.24.1\n",
      "cutadapt/1.10                    perl/5.24.1-thread-multi\n",
      "cutadapt/1.15                    perl/5.26.0\n",
      "data-registry/1.0                perl/5.26.0-thread-multi\n",
      "data-registry/1.0-dev            perl/5.28.0\n",
      "data-registry/1.0-test           perl/5.28.0-thread-multi\n",
      "deepvariant/0.7.2-py27h5d9141f_1 perl/5.8.9\n",
      "diamond/0.9.24                   perl/5.8.9-thread-multi\n",
      "Disco/1.1-beta                   perl/bio-extras\n",
      "docker-compose/1.11.2            perl/bio-extras-5.24.0-legacy\n",
      "docker-compose/1.7.1             perl/core\n",
      "DWGSIM/0561546                   perl/ensembl\n",
      "ea-utils/1.04.807                perl/jupyterhub\n",
      "ea-utils/1.1.2-484               perlbrew/0.76\n",
      "ea-utils/1.1.2-537               pfr-python2/2.7.13\n",
      "ea-utils/1.1.2-806               pfr-python3/3.6.1\n",
      "eddypro-engine/5.1.1             pfr-python3/3.6.5\n",
      "eddypro-engine/5.2.0             pfr-python3/3.6.6\n",
      "eddypro-engine/5.2.1             phylip/3.696\n",
      "eddypro-engine/6.0.0             phylobayes/4.1c\n",
      "eddypro-engine/6.1.0             PhyloSift/1.0.1\n",
      "eddypro-engine/6.2.0             picard-tools/1.112\n",
      "eddypro-engine/6.2.1             picard-tools/1.79\n",
      "EffectorP/1.0                    picard-tools/2.10.1\n",
      "eg-all/91                        picard-tools/2.18.7\n",
      "eg-web-bacteria/91               picard-tools/2.2.4\n",
      "eg-web-common/91                 picard-tools/2.9.4\n",
      "eg-web-fungi/91                  picrust2/2.3.0\n",
      "eg-web-metazoa/91                pilon/1.18\n",
      "eg-web-plants/91                 pilon/1.20\n",
      "eg-web-protists/91               pilon/1.23\n",
      "eg-web-search/91                 platanus/1.2.4\n",
      "eg-web-testsuite/91              platanus_trim/1.0.7\n",
      "ehive/2.4                        Platypus/0.8.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "eigensoft/7.2.1                  plink/1.07\n",
      "eman2/2.3                        plink/1.90b6.5\n",
      "emboss/6.4.0                     plink2/v2.00a2lm\n",
      "emboss/6.5.7                     plncpro/1.1\n",
      "emboss/6.6.0                     popoolation2/1201\n",
      "emirge/0.61.1                    porechop/0.2.3\n",
      "ensembl/69                       portcullis/1.1.0\n",
      "ensembl/91                       portcullis/1.1.1\n",
      "ensembl/92                       portcullis/1.1.2\n",
      "ensembl/93                       powerPlant/core\n",
      "ensembl-analysis/69              powerPlant/locate\n",
      "ensembl-analysis/91              pplacer/v1.1.alpha17-6-g5cecf99\n",
      "ensembl-analysis/92              ppsPCP/1.0\n",
      "ensembl-analysis/93              primer3/2.2.3\n",
      "ensembl-api/69                   primer3/2.3.5\n",
      "ensembl-api/91                   prinseq-lite/0.20.4\n",
      "ensembl-api/92                   Prodigal/2.6.3\n",
      "ensembl-api/93                   proj/5.2.0\n",
      "ensembl-compara/69               prokka/1.11\n",
      "ensembl-compara/91               prokka/1.13\n",
      "ensembl-compara/92               purge_haplotigs/1.1.0\n",
      "ensembl-compara/93               qiime/1.9.1\n",
      "ensembl-funcgen/69               qiime2/2018.11\n",
      "ensembl-funcgen/91               qiime2/2018.2\n",
      "ensembl-funcgen/92               qiime2/2019.1\n",
      "ensembl-funcgen/93               qiime2/2019.10\n",
      "ensembl-genebuild/69             qiime2/2019.1-picrust2\n",
      "ensembl-genebuild/91             qiime2/2019.4\n",
      "ensembl-genebuild/92             qiime2/2019.7\n",
      "ensembl-genebuild/93             qiime2/2019.7-picrust2\n",
      "ensemblgenomes-api/91            qualimap/v2.2.1\n",
      "ensembl-git-tools/d55068a        quast/4.2\n",
      "ensembl-hive/69                  R/3.3.0\n",
      "ensembl-hive/91                  R/3.3.1\n",
      "ensembl-hive/92                  R/3.3.3\n",
      "ensembl-hive/93                  R/3.4.0\n",
      "ensembl-hive-openlava/69         R/3.4.1\n",
      "ensembl-hive-openlava/91         R/3.4.2\n",
      "ensembl-hive-openlava/92         R/3.4.3\n",
      "ensembl-hive-openlava/93         R/3.4.4\n",
      "ensembl-io/69                    R/3.5.0\n",
      "ensembl-io/91                    R/3.5.2\n",
      "ensembl-io/92                    R/3.5.3\n",
      "ensembl-io/93                    R/3.6.1\n",
      "ensembl-killlist/69              racon/1.3.0\n",
      "ensembl-killlist/91              racon/1.3.1\n",
      "ensembl-killlist/92              racon/1.3.2\n",
      "ensembl-killlist/93              ragoo/1.01\n",
      "ensembl-metadata/69              ragoo/1.02\n",
      "ensembl-metadata/91              RAILS/v1.2\n",
      "ensembl-metadata/92              RANBOW/1\n",
      "ensembl-metadata/93              RANBOW/1-temp\n",
      "ensembl-notebook/latest          RECON/1.08\n",
      "ensembl-orm/69                   REPCLASS/1.0.1\n",
      "ensembl-orm/91                   RepeatMasker-open/3-3-0\n",
      "ensembl-orm/92                   RepeatMasker-open/4-0-3\n",
      "ensembl-orm/93                   RepeatMasker-open/4-0-5\n",
      "ensembl-pipeline/69              RepeatModeler/1.0.8\n",
      "ensembl-pipeline/91              RepeatModeler-open/1.0.11\n",
      "ensembl-pipeline/92              RepeatScout/1.0.5\n",
      "ensembl-pipeline/93              REPET/2.5\n",
      "ensembl-production/69            rnacocktail/0.2.1\n",
      "ensembl-production/91            RSEM/1.3.0\n",
      "ensembl-production/92            rtg-tools/3.7.1-eb13bbb\n",
      "ensembl-production/93            ruby/2.1.9\n",
      "ensembl-taxonomy/69              ruby/2.2.7\n",
      "ensembl-taxonomy/91              ruby/2.3.4\n",
      "ensembl-taxonomy/92              ruby/2.4.1\n",
      "ensembl-taxonomy/93              sabre/1.000\n",
      "ensembl-tools/69                 Salmon/0.8.2\n",
      "ensembl-tools/91                 Salmon/0.9.1\n",
      "ensembl-tools/92                 salsa/v2.2\n",
      "ensembl-tools/93                 samtools/0.1.18\n",
      "ensembl-variation/69             samtools/0.1.19\n",
      "ensembl-variation/91             samtools/1.1\n",
      "ensembl-variation/92             samtools/1.2\n",
      "ensembl-variation/93             samtools/1.3.1\n",
      "ensembl-vep/69                   samtools/1.7\n",
      "ensembl-vep/91                   samtools/1.9\n",
      "ensembl-vep/92                   ScanIndel/1.2dev\n",
      "ensembl-vep/93                   SDhap/56d7678\n",
      "e-PCR/2.3.9                      seqkit/0.7.0\n",
      "ericscript/0.5.5                 seqtk/1.2\n",
      "e-utilities/5.40                 sga/0.10.10-2\n",
      "EVidenceModeler/1.1.1            sga/0.10.13\n",
      "evigene/02-11-2017               sga/0.10.15\n",
      "evigene/18-01-2018               sga/0.9.19\n",
      "evigene/18-09-2013               sga/0.9.35\n",
      "exonerate/2.4.0                  SGSGeneLoss/0.1\n",
      "FALCON/1.7.4                     singularity/2.4.2\n",
      "fasta/35.4.12                    singularity/2.5.1\n",
      "fasta/36.3.8g                    singularity/3\n",
      "fast-gbs/2016.12.14              SLiM/3.1\n",
      "fast-gbs/2017.01.12              smalt/0.7.6\n",
      "fastp/0.12.5                     snakemake/5.10.0\n",
      "fastp/0.19.4                     snap/1.0beta.18\n",
      "fastp/0.20.0                     snapr/0.16alpha.51\n",
      "FastQC/0.11.1                    sniffles/1.0.11\n",
      "FastQC/0.11.2                    SnpEff/4.3t\n",
      "FastQC/0.11.7                    SNPhylo/20140701\n",
      "fastq_screen/v0.11.1             SNPhylo/20160204\n",
      "fastq_screen/v0.14.0             SOAPdenovo/1.05\n",
      "fastq_screen/v0.4.4              SOAPdenovo/2.04\n",
      "fastq_screen/v0.5.2              SOAPdenovo-Trans/1.03\n",
      "faststructure/1.0                SolexaQA/3.1.5\n",
      "FastTree/2.1.10                  sortmerna/2.1\n",
      "fastx_toolkit/0.0.13             sortmerna/3.0.3\n",
      "FGMP/1.0                         SPAdes/3.10.1\n",
      "FLASH/1.2.11                     SPAdes/3.11.1\n",
      "flye/2.4.2                       SPAdes/3.12.0\n",
      "flye/2.5                         SPAdes/3.13.0\n",
      "freebayes/1.1.0                  SPAdes/3.7.1\n",
      "freebayes/1.3.1                  SqueezeMeta/0.4.4\n",
      "freebayes/v1.0.2-16-gd466dde     SqueezeMeta/1.0.0-beta\n",
      "freetype/2.9                     sratoolkit/2.8.2.1\n",
      "GapFiller/v1.10                  SSPACE/2.0\n",
      "GARM/0.7.4                       stacks/1.40\n",
      "gatk/1.0                         stacks/1.40_withbam\n",
      "gatk/3.7                         stacks/2.0\n",
      "gatk/3.8.0                       stacks/2.1\n",
      "Gblocks/0.91b                    stacks/2.2\n",
      "gdal/2.4.0                       stamp/2.1.3\n",
      "GEM/20130406-045632              standard-RAxML/8.2.11\n",
      "GeMoMa/1.4.2                     standard-RAxML/8.2.9\n",
      "geneid/v1.4.4                    STAR/2.4.1d\n",
      "GeneMark/4.21                    STAR/2.4.2a\n",
      "GeneMark/4.33                    STAR/2.5.2a\n",
      "GeneMark/4.56                    STAR/2.5.2b\n",
      "genomethreader/1.7.0             STAR/2.5.3a\n",
      "genometools/1.5.10               STAR/2.6.1a\n",
      "getorganelle/1.6.2e              STAR/2.6.1d\n",
      "gffread/0.9.6                    STAR-Fusion/0.8.0\n",
      "git/2.11.0                       stringtie/1.3.3b.Linux_x86_64\n",
      "git/2.12.2                       subread/1.4.5\n",
      "git/2.14.1                       subread/1.5.3\n",
      "git/2.21.0                       supernova/1.2.2\n",
      "git-annex/6.20160613-g35dbe35    survivor/1.0.5\n",
      "gitflow/15aab26                  SV2/1.4.3.2\n",
      "git-lfs/2.3.0                    svaba/1.1.3\n",
      "globus-cli/1.2.2                 svim/1.2.0\n",
      "globus-cli/1.6.0                 svtools/0.3.2\n",
      "gmap/2016-06-09                  swan/3516c2f\n",
      "gmap/2017-06-20                  tabix/0.2.6\n",
      "gmap/2017-09-30                  TargQC/1.4.4\n",
      "gmap/2018-03-01                  tasr/1.6.2\n",
      "gmap/2018-03-25                  tassel/3.0\n",
      "gmap/2018-05-11                  tassel/4.0\n",
      "gmap/2018-05-30                  tassel/5.0\n",
      "gmato/v1.2                       tassel/5.2.28\n",
      "GMcloser/1.6.2                   taxmaps/0.2.1\n",
      "go/1.9                           texlive/20151117\n",
      "GroIMP/1.5                       tigmint/1.1.2\n",
      "gromacs/2018                     tophat/2.0.13\n",
      "gromacs/2018.1                   tophat/2.1.1\n",
      "gromacs/2018.2                   totalview/2017.2.11\n",
      "gromacs/2018.3                   TransDecoder/3.0.1\n",
      "gromacs/2018.4                   TransDecoder/5.0.1\n",
      "gromacs/2018.5                   transrate/1.0.3\n",
      "gromacs/2019                     trf/4.07\n",
      "guppy/3.2.4                      trim_galore/0.4.1\n",
      "guppy/3.4.4                      trim_galore/0.4.3\n",
      "hagfish/05d08d1                  Trimmomatic/0.36\n",
      "hagfish/1_May_2013               trinityrnaseq/2.0.6\n",
      "hapcol/97d4a5e                   trinityrnaseq/2.2.0\n",
      "hapcompass/0.8.2                 trinityrnaseq/2.3.2\n",
      "HapCUT2/0121162                  trinityrnaseq/2.4.0\n",
      "Haplosim/ae652a2                 trinityrnaseq/2.5.1\n",
      "Haplosim/bd39c6f                 trinityrnaseq/2.6.5\n",
      "HapTree/1.0                      trinotate/3.0.2\n",
      "hdf5/1.8.17                      tRNAscan-SE/1.3.1\n",
      "HiC-pro/2.11.1                   tRNAscan-SE/2.0\n",
      "hicup/0.5.9                      ucsc-pslcdnafilter/324\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hicup/0.7.2                      unixODBC/2.3.0\n",
      "hisat2/2.1.0                     vcflib/50a3229\n",
      "hmmer/3.0                        vcflib/586c5ae\n",
      "hmmer/3.1b1                      vcflib/73b7f13\n",
      "hmmer/3.1b2                      vcftools/0.1.12\n",
      "HTSeq/0.6.1                      vcftools/0.1.14\n",
      "HTSeq/0.7.2                      velvet/1.2.10\n",
      "HTSeq/0.9.1                      VersioningService/69\n",
      "htslib/1.3.1                     VersioningService/91\n",
      "htslib/1.7                       VersioningService/92\n",
      "htslib/1.9                       VersioningService/93\n",
      "hub/2.8.4                        vg/1.10.0\n",
      "humann2/0.11.1                   vg/1.11.0\n",
      "iCommands/4.1.8                  vg/1.12.0\n",
      "idba/a1bafe6                     vg/1.12.1\n",
      "IGV/2.3.93                       vg/1.13.0\n",
      "infernal/1.1.2                   vg/1.7.0\n",
      "iqtree/1.6.11                    vg/1.8.0\n",
      "JAGS/4.0.0                       vg/1.9.0\n",
      "JAGS/4.2.0                       ViennaRNA/2.2.2\n",
      "java/1.7.0                       vsearch/2.3.2\n",
      "java/1.8.0                       wgs/8.3rc2\n",
      "jcvi/0.7.3                       whatshap/0.14.1-py36_0\n",
      "jcvi/0.7.7                       wise/2.4.1\n",
      "jcvi/0.8.12                      wkhtmlox/0.12.3\n",
      "jellyfish/1.1.10                 wtdbg2/2.2\n",
      "jellyfish/2.1.3\n"
     ]
    }
   ],
   "source": [
    "module avail"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Function:\n",
    "Download teleost lineage data of common orthologs and untar/gz the downloaded file.\n",
    "    \n",
    "Output:\n",
    "1. actinopterygii_odb10.2019-11-20(folder with lineage data)\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2020-02-20 15:40:35--  https://busco-data.ezlab.org/v4/data/lineages/actinopterygii_odb10.2019-11-20.tar.gz\n",
      "Resolving busco-data.ezlab.org (busco-data.ezlab.org)... 129.194.96.125\n",
      "Connecting to busco-data.ezlab.org (busco-data.ezlab.org)|129.194.96.125|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 172780820 (165M) [application/octet-stream]\n",
      "Saving to: ‘actinopterygii_odb10.2019-11-20.tar.gz’\n",
      "\n",
      "100%[======================================>] 172,780,820 10.6MB/s   in 18s    \n",
      "\n",
      "2020-02-20 15:40:56 (9.11 MB/s) - ‘actinopterygii_odb10.2019-11-20.tar.gz’ saved [172780820/172780820]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#download teleost lineage data\n",
    "wget https://busco-data.ezlab.org/v4/data/lineages/actinopterygii_odb10.2019-11-20.tar.gz\n",
    "tar xzf actinopterygii_odb10.2019-11-20.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. config(folder with augustus prerequisites from the software/bioinformatics path)\n",
    "\n",
    "Function:\n",
    "Copy the augustus-3.3.1 config recursively into the 'Diagnostics_assembly' folder working directory\n",
    "in order to call a local AUGUSTUS_CONFIG_PATH \n",
    "later on, because of associated rights involved when running BUSCO related to augustus.\n",
    "    \n",
    "Output:\n",
    "1. conig(folder with augustus prerequisites in the 'Diagnostics assembly' folder working directory)\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#config path dl to local since lacking rights for revision needed for busco\n",
    "cp -r /software/bioinformatics/augustus-3.3.1/config /powerplant/workspace/hramzr/github/Trevally/PacBio/Diagnostics_assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. fetreval.fasta.contigs.fasta(FASTA format assembly of the canu_filtered run)\n",
    "\n",
    "Function:\n",
    "Call BUSCO stats which are needed for validation purposes of assemblies based on orthologues.\n",
    "    \n",
    "Output:\n",
    "1. canu_filtered(folder with BUSCO output stats)\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <335792> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "#run busco and change august config path to be changed for lineage\n",
    "# https://gitlab.com/ezlab/busco/issues/15\n",
    "module purge\n",
    "module load perl/bio-extras\n",
    "module load augustus/3.2.2\n",
    "module load pfr-python3/3.6.1\n",
    "module load hmmer/3.1b2\n",
    "module load blast/2.2.26 \n",
    "module load BUSCO/v3.0.2     \n",
    "\n",
    "logdir=/workspace/hramzr/github/Trevally/PacBio/log\n",
    "genomecanu=/workspace/hramzr/github/Trevally/PacBio/canu_filtered/fetreval.fasta.contigs.fasta\n",
    "export AUGUSTUS_CONFIG_PATH=\"/workspace/hramzr/github/Trevally/PacBio/Diagnostics_assembly/config\"\n",
    "bsub -o $logdir/busco.out -e $logdir/busco.err -n 8 -J \"BUSCO\" \\\n",
    "\"run_BUSCO.py -r -m genome -i $genomecanu -o canu_filtered -l actinopterygii_odb10\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. short_summary_OUTPUT.txt(Text file with short summary output of the BUSCO run on the assembly)\n",
    "\n",
    "Function:\n",
    "Create an R script that plots the data from the short_summar_OUTPUT.txt file by running it in a python script called \n",
    "\"generate_plot.py\".\n",
    "    \n",
    "Output:\n",
    "1. busco_figure.R (R script that can plot the short_summary_OUTPUT data)\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Conflict with existing Conda environment. Please unload/deactivate pfr-python3-3.6.1\n",
      "INFO\t****************** Start plot generation at 02/24/2020 11:53:30 ******************\n",
      "INFO\tLoad data ...\n",
      "INFO\tLoaded short_summary_OUTPUT.txt successfully\n",
      "INFO\tGenerate the R code ...\n",
      "INFO\tYou chose not to run R\n",
      "INFO\tPlot generation done. Total running time: 0.004861116409301758 seconds\n",
      "INFO\tResults written in /workspace/hramzr/github/Trevally/PacBio/Diagnostics_assembly/run_OUTPUT/BUSCO_plots/\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#create plot dir and generate R script to plot results \n",
    "module load pfr-python3/3.6.1\n",
    "module load R/3.4.3\n",
    "module load BUSCO/v3.0.2   \n",
    "mkdir BUSCO_plots\n",
    "cp short_summary_OUTPUT.txt BUSCO_plots/.\n",
    "\n",
    "plt=/software/bioinformatics/BUSCO_v3.0.2/scripts/generate_plot.py\n",
    "wkdir=/workspace/hramzr/github/Trevally/PacBio/Diagnostics_assembly/run_OUTPUT/BUSCO_plots/\n",
    "python3 $plt -wd $wkdir  --no_r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#######################################\n",
    "Input:\n",
    "1. busco_figure.R (R script that generates a png from BUSCO output data)\n",
    "\n",
    "Function:\n",
    "Run the script 'busco_figure.R' to create a BUSCO data plot.\n",
    "\n",
    "Output:\n",
    "1. busco_figure.png (PNG file that contains a visualization of the short_summary_OUTPUT.txt file)\n",
    "#######################################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <317299> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "#run r script and produce plots\n",
    "plt=/workspace/hramzr/github/Trevally/PacBio/Diagnostics_assembly/run_OUTPUT/BUSCO_plots/busco_figure.R \n",
    "module load R/3.4.3 \n",
    "bsub -n 8 -e Rerror.err \"Rscript $plt\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
